// @shopify/hydrogen/dist/production/index.cjs
'use strict';

var hydrogenReact = require('@shopify/hydrogen-react');
var react = require('react');
var react$1 = require('@remix-run/react');
var jsxRuntime = require('react/jsx-runtime');
var cookie = require('worktop/cookie');
var Sa = require('content-security-policy-builder');
var serverRuntime = require('@remix-run/server-runtime');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var Sa__default = /*#__PURE__*/_interopDefault(Sa);

var Lo=Object.defineProperty;var No=(e,t)=>()=>(e&&(t=e(e=0)),t);var Vo=(e,t)=>{for(var r in t)Lo(e,r,{get:t[r],enumerable:!0});};var kr={};Vo(kr,{default:()=>_r,logSeoTags:()=>Ur});function _r({headTags:e}){return Ur(e),null}function Ur(e){console.log(" "),console.log("%cSEO Meta Tags",`${pn}`),console.log(" "),e.forEach(t=>{if(t.tag==="script"){if(console.log("%c\u2022 JSON LD ",kt),t.children)try{console.table(JSON.parse(t.children),["name","content"]);}catch{console.log(t.children);}}else {if(console.log(`%c\u2022 ${t.tag} `,kt),t.children)if(typeof t.children=="string")console.log(`\u21B3 ${t.children}`);else try{Object.entries(JSON.parse(t.children)).map(([r,o])=>console.log(`\u21B3 ${o}`));}catch{console.log(t.children);}if(t.props.property==="og:image:url"){let r=t.props.content;dn(r).then(o=>{let n=`font-size: 400px; padding: 10px; background: white url(${o}) no-repeat center; background-size: contain;`;console.log("%c\u2022 Share image preview",kt),console.log("%c  ",n),console.log(`\u21B3 ${r}`);}).catch(o=>{console.error(o);});}Object.entries(t.props).map(([r,o])=>{console.log(`\u21B3 ${r} \u2192 ${o}`);});}console.log(" ");});}async function dn(e){let o=await(await(await fetch(e)).blob()).arrayBuffer();return `data:image/png;base64,${ln(o)}`}function ln(e){let t="",r=new Uint8Array(e),o=r.byteLength;for(let n=0;n<o;n++)t+=String.fromCharCode(r[n]);return btoa(t)}var kt,pn,Lr=No(()=>{kt="text-transform: uppercase;",pn="text-transform: uppercase; font-weight: bold; text-transform: uppercase;font-weight: bold";});var Ve="public",Mo="private",pt="no-store",dr={maxAge:"max-age",staleWhileRevalidate:"stale-while-revalidate",sMaxAge:"s-maxage",staleIfError:"stale-if-error"};function Pe(e){let t=[];return Object.keys(e).forEach(r=>{r==="mode"?t.push(e[r]):dr[r]&&t.push(`${dr[r]}=${e[r]}`);}),t.join(", ")}function dt(){return {mode:pt}}function lt(e){if(e?.mode&&e?.mode!==Ve&&e?.mode!==Mo)throw Error("'mode' must be either 'public' or 'private'")}function ce(e){return lt(e),{mode:Ve,maxAge:1,staleWhileRevalidate:9,...e}}function yt(e){return lt(e),{mode:Ve,maxAge:3600,staleWhileRevalidate:82800,...e}}function le(e){return lt(e),{mode:Ve,maxAge:1,staleWhileRevalidate:86399,...e}}function ft(e){return e}function K(e){return String(e).includes("__proto__")?JSON.parse(e,Ho):JSON.parse(e)}function Ho(e,t){if(e!=="__proto__")return t}function Me(e,t){return e&&t?{...e,...t}:e||le()}function mt(e){return Pe(Me(e))}async function Fo(e,t){if(!e)return;let r=await e.match(t);if(!r){return}return r}async function qo(e,t,r,o){if(!e)return;let n=Me(o),a=mt(Me(n,{maxAge:(n.maxAge||0)+(n.staleWhileRevalidate||0)})),s=mt(Me(n));r.headers.set("cache-control",a),r.headers.set("real-cache-control",s),r.headers.set("cache-put-date",String(Date.now())),await e.put(t,r);}async function $o(e,t){e&&await e.delete(t);}function Bo(e,t){let r=e.headers.get("real-cache-control"),o=0;if(r){let a=r.match(/max-age=(\d*)/);a&&a.length>1&&(o=parseFloat(a[1]));}return [(Date.now()-Number(t))/1e3,o]}function Go(e,t){let r=t.headers.get("cache-put-date");if(!r)return !1;let[o,n]=Bo(t,r),a=o>n;return a}var He={get:Fo,set:qo,delete:$o,generateDefaultCacheControlHeader:mt,isStale:Go};function Ae(e){return `https://shopify.dev/?${e}`}function Qo(e){return e||le()}async function lr(e,t){if(!e)return;let r=Ae(t),o=new Request(r),n=await He.get(e,o);if(!n)return;let a=await n.text();try{return [K(a),n]}catch{return [a,n]}}async function yr(e,t,r,o){if(!e)return;let n=Ae(t),a=new Request(n),s=new Response(JSON.stringify(r));await He.set(e,a,s,Qo(o));}function fr(e,t){return He.isStale(new Request(Ae(e)),t)}function mr(e){let t=Array.isArray(e)?e:[e],r="";for(let o of t)o!=null&&(typeof o=="object"?r+=JSON.stringify(o):r+=o.toString());return encodeURIComponent(r)}var gt=new Set;async function Fe(e,t,{strategy:r=ce(),cacheInstance:o,shouldCacheResult:n=()=>!0,waitUntil:a,debugInfo:s}){let i=mr([...typeof e=="string"?[e]:e]),y=m=>{({displayName:m.displayName,url:m.response?.url,responseInit:{status:m.response?.status||0,statusText:m.response?.statusText||"",headers:Array.from(m.response?.headers.entries()||[])}});},d=void 0;if(!o||!r||r.mode===pt){let m=await t({addDebugData:y});return m}let p=m=>yr(o,i,{value:m,debugInfo:void 0},r),f=await lr(o,i);if(f&&typeof f[0]!="string"){let[{value:m,debugInfo:C},A]=f;let T=fr(i,A)?"STALE":"HIT";if(!gt.has(i)&&T==="STALE"){gt.add(i);let h=Promise.resolve().then(async()=>{let I=Date.now();try{let S=await t({addDebugData:y});n(S)&&(await p(S),d?.({result:S,cacheStatus:"PUT",overrideStartTime:I}));}catch(S){S.message&&(S.message="SWR in sub-request failed: "+S.message),console.error(S);}finally{gt.delete(i);}});a?.(h);}return m}let P=await t({addDebugData:y});if(n(P)){let m=Promise.resolve().then(async()=>{await p(P);});a?.(m);}return P}var Wo=["set-cookie","server-timing"];function jo(e,t){return [e,{status:t.status,statusText:t.statusText,headers:[...t.headers].filter(([r])=>!Wo.includes(r.toLowerCase()))}]}function gr([e,t]){return [e,new Response(e,t)]}async function qe(e,t,{cacheInstance:r,cache:o,cacheKey:n=[e,t],shouldCacheResponse:a,waitUntil:s,debugInfo:c,onRawHeaders:i}){return !o&&(!t.method||t.method==="GET")&&(o=ce()),Fe(n,async()=>{let u=await fetch(e,t);if(i?.(u.headers),!u.ok)return u;let l=await u.text().catch(()=>"");try{l&&(l=K(l));}catch{}return jo(l,u)},{cacheInstance:r,waitUntil:s,strategy:o??null,debugInfo:c,shouldCacheResult:u=>"ok"in u?!1:a(...gr(u))}).then(u=>"ok"in u?[null,u]:gr(u))}var $e="Custom-Storefront-Request-Group-ID",Be="X-Shopify-Storefront-Access-Token",ht="X-SDK-Variant",Ct="X-SDK-Variant-Source",St="X-SDK-Version",Pt="X-Shopify-Client-IP",At="X-Shopify-Client-IP-Sig",Ge="_sfapi_proxy",Qe="_server_tracking";function We(){return typeof crypto<"u"&&crypto.randomUUID?crypto.randomUUID():`weak-${Math.random().toString(16).substring(2)}`}var hr=new Set,re=e=>{hr.has(e)||(console.warn(e),hr.add(e));},Cr=new Set,Tt=e=>{Cr.has(e)||(console.error(new Error(e)),Cr.add(e));};var Te="2024.10.2";function ye(e){return e.replace(/\s*#.*$/gm,"").replace(/\s+/gm," ").trim()}var Ko=/(^|}\s)query[\s({]/im,Yo=/(^|}\s)mutation[\s({]/im;function je(e,t){if(!Ko.test(e))throw new Error(`[h2:error:${t}] Can only execute queries`)}function Ke(e,t){if(!Yo.test(e))throw new Error(`[h2:error:${t}] Can only execute mutations`)}var ue=class extends Error{locations;path;extensions;constructor(t,r={}){let n=(r.clientOperation?`[h2:error:${r.clientOperation}] `:"")+t+(r.requestId?` - Request ID: ${r.requestId}`:"");super(n),this.name="GraphQLError",this.extensions=r.extensions,this.locations=r.locations,this.path=r.path,this.stack=r.stack||void 0;try{this.cause=JSON.stringify({...typeof r.cause=="object"?r.cause:{},requestId:r.requestId});}catch{r.cause&&(this.cause=r.cause);}}get[Symbol.toStringTag](){return this.name}toString(){let t=`${this.name}: ${this.message}`;if(this.path)try{t+=` | path: ${JSON.stringify(this.path)}`;}catch{}if(this.extensions)try{t+=` | extensions: ${JSON.stringify(this.extensions)}`;}catch{}return t+=`
`,this.stack&&(t+=`${this.stack.slice(this.stack.indexOf(`
`)+1)}
`),t}toJSON(){return {name:"Error",message:""}}};function ve({url:e,response:t,errors:r,type:o,query:n,queryVariables:a,ErrorConstructor:s=Error,client:c="storefront"}){let i=(typeof r=="string"?r:r?.map?.(l=>l.message).join(`
`))||`URL: ${e}
API response error: ${t.status}`,u=new ue(i,{query:n,queryVariables:a,cause:{errors:r},clientOperation:`${c}.${o}`,requestId:t.headers.get("x-request-id")});throw new s(u.message,{cause:u.cause})}function fe(e,t={}){let r=new Error,o=(n,a="Error")=>{let s=(r.stack??"").split(`
`).slice(3+(t.stackOffset??0)).join(`
`).replace(/ at loader(\d+) \(/,(c,i)=>c.replace(i,""));return `${a}: ${n}
`+s};return e.then(n=>{if(n?.errors&&Array.isArray(n.errors)){let a=typeof t.logErrors=="function"?t.logErrors:()=>t.logErrors??!1;n.errors.forEach(s=>{s&&(s.stack=o(s.message,s.name),a(s)&&console.error(s));});}return n}).catch(n=>{throw n&&(n.stack=o(n.message,n.name)),n})}var Q=void 0;function B(e,t){return vt(e.headers,t)}function vt(e,t){let r=e?.get?.(t)??e?.[t];return typeof r=="string"?r:null}function oe(e){return {requestId:e?B(e,"request-id"):void 0,purpose:e?B(e,"purpose"):void 0}}var It=/^\/api\/(unstable|2\d{3}-\d{2})\/graphql\.json$/,Et=e=>{try{return new URL(e,"http://e.c").pathname}catch{return "/"}};function Rt(e,t){return t.reduce((r,o)=>{let n=e(o);return n&&r.push([o,n]),r},[])}function zo(e){return Object.entries(e).map(([t,r])=>r?`${t};desc=${r}`:void 0).filter(Boolean).join(", ")}function Ie(e,t){let r=typeof t=="string"?t:zo(t);r&&e.headers.append("Server-Timing",r);}var Jo=["_y","_s","_cmp"];function Sr(e){let t={};if(!e)return t;let r=new RegExp(`\\b(${Jo.join("|")});desc="?([^",]+)"?`,"g"),o;for(;(o=r.exec(e))!==null;)t[o[1]]=o[2];return t}function Pr(e){if(typeof window>"u")return !1;try{return !!window.performance.getEntriesByType("navigation")[0]?.serverTiming?.some(r=>r.name===e)}catch{return !1}}function Ar(){return Pr(Ge)}function Tr(){return Pr(Qe)}var tn={language:"EN",country:"US"};function br(e){let {storefrontHeaders:t,cache:r,waitUntil:o,i18n:n,storefrontId:a,logErrors:s=!0,...c}=e,{getPublicTokenHeaders:u,getPrivateTokenHeaders:l,getStorefrontApiUrl:y,getShopifyDomain:g}=hydrogenReact.createStorefrontClient(c),p=(c.privateStorefrontToken?l:u)({contentType:"json",buyerIp:t?.buyerIp||""});t?.buyerIp&&(p[Pt]=t.buyerIp),t?.buyerIpSig&&(p[At]=t.buyerIpSig),p[$e]=t?.requestGroupId||We(),a&&(p[hydrogenReact.SHOPIFY_STOREFRONT_ID_HEADER]=a),(p["user-agent"]=`Hydrogen ${Te}`);let f=t?.cookie??"";f&&(p.cookie=f);let P,m;if(!/\b_shopify_(analytics|marketing)=/.test(f)){let h=f.match(/\b_shopify_y=([^;]+)/)?.[1],I=f.match(/\b_shopify_s=([^;]+)/)?.[1];h&&(p[hydrogenReact.SHOPIFY_STOREFRONT_Y_HEADER]=h),I&&(p[hydrogenReact.SHOPIFY_STOREFRONT_S_HEADER]=I),P=h??We(),m=I??We(),p[hydrogenReact.SHOPIFY_UNIQUE_TOKEN_HEADER]=P,p[hydrogenReact.SHOPIFY_VISIT_TOKEN_HEADER]=m;}let C,A=JSON.stringify({"content-type":p["content-type"],"user-agent":p["user-agent"],[ht]:p[ht],[Ct]:p[Ct],[St]:p[St],[Be]:p[Be]});async function T({query:h,mutation:I,variables:S,cache:M,headers:U=[],storefrontApiVersion:Z,displayName:ct,stackInfo:Le}){let ut=U instanceof Headers?Object.fromEntries(U.entries()):Array.isArray(U)?Object.fromEntries(U):U,ee=h??I,v={...S};n&&(!S?.country&&/\$country/.test(ee)&&(v.country=n.country),!S?.language&&/\$language/.test(ee)&&(v.language=n.language));let E=y({storefrontApiVersion:Z}),b=JSON.stringify({query:ee,variables:v}),w={method:"POST",headers:{...p,...ut},body:b},W=[E,w.method,A,w.body],[te,_]=await qe(E,w,{cacheInstance:I?void 0:r,cache:M||le(),cacheKey:W,waitUntil:o,shouldCacheResponse:L=>!L?.errors,debugInfo:{requestId:w.headers[$e],displayName:ct,url:E,stackInfo:Le,graphql:b,purpose:t?.purpose},onRawHeaders:L=>{C??={setCookie:typeof L.getSetCookie=="function"?L.getSetCookie():[],serverTiming:L.get("server-timing")??""};}}),F={url:E,response:_,type:I?"mutation":"query",query:ee,queryVariables:v,errors:void 0};if(!_.ok){let L,G=te;try{G??=await _.text(),L=K(G);}catch{L=[{message:G??"Could not parse Storefront API response"}];}ve({...F,errors:L});}let{data:$,errors:k}=te,j=k?.map(({message:L,...G})=>new ue(L,{...G,clientOperation:`storefront.${F.type}`,requestId:_.headers.get("x-request-id"),queryVariables:v,query:ee}));return R($,j)}return {storefront:{query(h,I){h=ye(h),je(h,"storefront.query");let S=Rr?.(h);return fe(T({...I,query:h,stackInfo:Q?.(S)}),{stackOffset:S,logErrors:s})},mutate(h,I){h=ye(h),Ke(h,"storefront.mutate");let S=Rr?.(h);return fe(T({...I,mutation:h,stackInfo:Q?.(S)}),{stackOffset:S,logErrors:s})},cache:r,CacheNone:dt,CacheLong:yt,CacheShort:ce,CacheCustom:ft,generateCacheControlHeader:Pe,getPublicTokenHeaders:u,getPrivateTokenHeaders:l,getHeaders:()=>({...p}),getShopifyDomain:g,getApiUrl:y,i18n:n??tn,isStorefrontApiUrl(h){return It.test(Et(h.url??""))},async forward(h,I){let S=new Headers([...Rt(Z=>h.headers.get(Z),["accept","accept-encoding","accept-language","access-control-request-headers","access-control-request-method","content-type","content-length","cookie","origin","referer","user-agent",Be,hydrogenReact.SHOPIFY_UNIQUE_TOKEN_HEADER,hydrogenReact.SHOPIFY_VISIT_TOKEN_HEADER]),...Rt(Z=>p[Z],[Pt,At,hydrogenReact.SHOPIFY_STOREFRONT_ID_HEADER,$e])]);t?.buyerIp&&S.set("x-forwarded-for",t.buyerIp);let M=I?.storefrontApiVersion??Et(h.url).match(It)?.[1],U=await fetch(y({storefrontApiVersion:M}),{method:h.method,body:h.body,headers:S});return new Response(U.body,U)},setCollectedSubrequestHeaders:h=>{if(C)for(let U of C.setCookie)h.headers.append("Set-Cookie",U);let I=Sr(C?.serverTiming),S=h.headers.get("content-type")?.startsWith("text/html");Ie(h,{...S?{_y:P,_s:m}:void 0,...I}),S&&C&&C.setCookie.length>1&&I?._y&&I?._s&&I?._cmp&&Ie(h,{[Qe]:"1"});}}}}var Rr=void 0;function R(e,t){return {...e,...t&&{errors:t}}}function rn(e){let{cache:t,waitUntil:r,request:o}=e;return {run:({cacheKey:n,cacheStrategy:a,shouldCacheResult:s},c)=>Fe(n,c,{shouldCacheResult:s,strategy:a,cacheInstance:t,waitUntil:r,debugInfo:{...oe(o),stackInfo:Q?.()}}),fetch:(n,a,s)=>qe(n,a??{},{waitUntil:r,cacheKey:[n,a],cacheInstance:t,debugInfo:{url:n,...oe(o),stackInfo:Q?.(),displayName:s?.displayName},cache:s.cacheStrategy,...s}).then(([c,i])=>({data:c,response:i}))}}var bt=class{#e;constructor(){this.#e=new Map;}add(t){throw new Error("Method not implemented. Use `put` instead.")}addAll(t){throw new Error("Method not implemented. Use `put` instead.")}matchAll(t,r){throw new Error("Method not implemented. Use `match` instead.")}async put(t,r){if(t.method!=="GET")throw new TypeError("Cannot cache response to non-GET request.");if(r.status===206)throw new TypeError("Cannot cache response to a range request (206 Partial Content).");if(r.headers.get("vary")?.includes("*"))throw new TypeError("Cannot cache response with 'Vary: *' header.");this.#e.set(t.url,{body:new Uint8Array(await r.arrayBuffer()),status:r.status,headers:[...r.headers],timestamp:Date.now()});}async match(t){if(t.method!=="GET")return;let r=this.#e.get(t.url);if(!r)return;let{body:o,timestamp:n,...a}=r,s=new Headers(a.headers),c=s.get("cache-control")||s.get("real-cache-control")||"",i=parseInt(c.match(/max-age=(\d+)/)?.[1]||"0",10),u=parseInt(c.match(/stale-while-revalidate=(\d+)/)?.[1]||"0",10),l=(Date.now()-n)/1e3;if(l>i+u){this.#e.delete(t.url);return}let g=l>i;return s.set("cache",g?"STALE":"HIT"),s.set("date",new Date(n).toUTCString()),new Response(o,{status:a.status??200,headers:s})}async delete(t){return this.#e.has(t.url)?(this.#e.delete(t.url),!0):!1}keys(t){let r=[];for(let o of this.#e.keys())(!t||t.url===o)&&r.push(new Request(o));return Promise.resolve(r)}};function Ye(e){if(!e)return;let{pathname:t,search:r}=new URL(e),o=t+r,n=new URLSearchParams(r),a=n.get("return_to")||n.get("redirect");if(a){if(Or(e,a))return a;console.warn(`Cross-domain redirects are not supported. Tried to redirect from ${o} to ${a}`);}}function Or(e,t){try{return new URL(e).origin===new URL(t,e).origin}catch{return !1}}function wt({requestUrl:e,defaultUrl:t,redirectUrl:r}){let o=e,n=wr(e,t),a=r?wr(e,r):n;return Or(e,a.toString())?a.toString():(console.warn(`Cross-domain redirects are not supported. Tried to redirect from ${o} to ${a}. Default url ${n} is used instead.`),n.toString())}function wr(e,t){return on(t)?new URL(t):new URL(t,new URL(e).origin)}function on(e){try{return new URL(e),!0}catch{return !1}}async function nn(e){let{storefront:t,request:r,noAdminRedirect:o,matchQueryParams:n,response:a=new Response("Not Found",{status:404})}=e,s=new URL(r.url),{pathname:c,searchParams:i}=s,u=i.has("_data");i.delete("redirect"),i.delete("return_to"),i.delete("_data");let l=(n?s.toString().replace(s.origin,""):c).toLowerCase();if(s.pathname==="/admin"&&!o)return xt(`${t.getShopifyDomain()}/admin`,u,i,n);try{let{urlRedirects:y}=await t.query(an,{variables:{query:"path:"+l.replace(/\/+$/,"")}}),g=y?.edges?.[0]?.node?.target;if(g)return xt(g,u,i,n);let d=Ye(r.url);if(d)return xt(d,u,i,n)}catch(y){console.error(`Failed to fetch redirects from Storefront API for route ${l}`,y);}return a}var Ot="https://example.com";function xt(e,t,r,o){let n=new URL(e,Ot);if(!o)for(let[a,s]of r)n.searchParams.append(a,s);return t?new Response(null,{status:200,headers:{"X-Remix-Redirect":n.toString().replace(Ot,""),"X-Remix-Status":"301"}}):new Response(null,{status:301,headers:{location:n.toString().replace(Ot,"")}})}var an=`#graphql
  query redirects($query: String) {
    urlRedirects(first: 1, query: $query) {
      edges {
        node {
          target
        }
      }
    }
  }
`;var sn=async function({request:t,context:r}){let o=r.storefront,n=r.customerAccount,a=new URL(t.url);if(!o)throw new Error("GraphiQL: Hydrogen's storefront client must be injected in the loader context.");let s={};if(o){let u="X-Shopify-Storefront-Access-Token";s.storefront={name:"Storefront API",authHeader:u,accessToken:o.getPublicTokenHeaders()[u],apiUrl:o.getApiUrl(),icon:"SF"};}if(n){let u=await(await fetch(a.origin+"/graphiql/customer-account.schema.json")).json(),l=await n.getAccessToken();u&&(s["customer-account"]={name:"Customer Account API",value:u,authHeader:"Authorization",accessToken:l,apiUrl:n.getApiUrl(),icon:"CA"});}let c="https://avatars.githubusercontent.com/u/12972006?s=48&v=4",i=String.raw;return new Response(i`
      <!DOCTYPE html>
      <html lang="en">
        <head>
          <title>GraphiQL</title>
          <link rel="icon" type="image/x-icon" href="${c}" />
          <style>
            body {
              height: 100%;
              margin: 0;
              width: 100%;
              overflow: hidden;
              background-color: hsl(219, 29%, 18%);
            }

            #graphiql {
              height: 100vh;
            }

            #graphiql > .placeholder {
              color: slategray;
              width: fit-content;
              margin: 40px auto;
              font-family: Arial;
            }

            .graphiql-api-toolbar-label {
              position: absolute;
              bottom: -6px;
              right: -4px;
              font-size: 8px;
            }
          </style>

          <script
            crossorigin
            src="https://unpkg.com/react@18/umd/react.development.js"
          ></script>
          <script
            crossorigin
            src="https://unpkg.com/react-dom@18/umd/react-dom.development.js"
          ></script>
          <link
            rel="stylesheet"
            href="https://unpkg.com/graphiql@3/graphiql.min.css"
          />
          <link
            rel="stylesheet"
            href="https://unpkg.com/@graphiql/plugin-explorer/dist/style.css"
          />
        </head>

        <body>
          <div id="graphiql">
            <div class="placeholder">Loading GraphiQL...</div>
          </div>

          <script
            src="https://unpkg.com/graphiql@3/graphiql.min.js"
            type="application/javascript"
            crossorigin="anonymous"
          ></script>
          <script
            src="https://unpkg.com/@graphiql/plugin-explorer/dist/index.umd.js"
            type="application/javascript"
            crossorigin="anonymous"
          ></script>

          <script>
            const windowUrl = new URL(document.URL);
            const startingSchemaKey =
              windowUrl.searchParams.get('schema') || 'storefront';

            let query = '{ shop { name } }';
            if (windowUrl.searchParams.has('query')) {
              query = decodeURIComponent(
                windowUrl.searchParams.get('query') ?? query,
              );
            }

            // Prettify query
            query = GraphiQL.GraphQL.print(GraphiQL.GraphQL.parse(query));

            let variables;
            if (windowUrl.searchParams.has('variables')) {
              variables = decodeURIComponent(
                windowUrl.searchParams.get('variables') ?? '',
              );
            }

            // Prettify variables
            if (variables) {
              variables = JSON.stringify(JSON.parse(variables), null, 2);
            }

            const schemas = ${JSON.stringify(s)};
            let lastActiveTabIndex = -1;
            let lastTabAmount = -1;

            const root = ReactDOM.createRoot(
              document.getElementById('graphiql'),
            );

            root.render(React.createElement(RootWrapper));

            const TAB_STATE_KEY = 'graphiql:tabState';
            const storage = {
              getTabState: () =>
                JSON.parse(localStorage.getItem(TAB_STATE_KEY)),
              setTabState: (state) =>
                localStorage.setItem(TAB_STATE_KEY, JSON.stringify(state)),
            };

            let nextSchemaKey;

            function RootWrapper() {
              const [activeSchema, setActiveSchema] =
                React.useState(startingSchemaKey);

              const schema = schemas[activeSchema];
              if (!schema) {
                throw new Error('No schema found for ' + activeSchema);
              }

              const keys = Object.keys(schemas);

              return React.createElement(
                GraphiQL,
                {
                  fetcher: GraphiQL.createFetcher({
                    url: schema.apiUrl,
                    headers: {[schema.authHeader]: schema.accessToken},
                  }),
                  defaultEditorToolsVisibility: true,
                  query,
                  variables,
                  schema: schema.value,
                  plugins: [GraphiQLPluginExplorer.explorerPlugin()],
                  onTabChange: (state) => {
                    const {activeTabIndex, tabs} = state;
                    const activeTab = tabs[activeTabIndex];

                    if (
                      activeTabIndex === lastActiveTabIndex &&
                      lastTabAmount === tabs.length
                    ) {
                      if (
                        nextSchemaKey &&
                        activeTab &&
                        activeTab.schemaKey !== nextSchemaKey
                      ) {
                        activeTab.schemaKey = nextSchemaKey;
                        nextSchemaKey = undefined;

                        // Sync state to localStorage. GraphiQL resets the state
                        // asynchronously, so we need to do it in a timeout.
                        storage.setTabState(state);
                        setTimeout(() => storage.setTabState(state), 500);
                      }

                      // React rerrendering, skip
                      return;
                    }

                    if (activeTab) {
                      if (!activeTab.schemaKey) {
                        // Creating a new tab
                        if (lastTabAmount < tabs.length) {
                          activeTab.schemaKey = activeSchema;
                          storage.setTabState(state);
                        }
                      }

                      const nextSchema = activeTab.schemaKey || 'storefront';

                      if (nextSchema !== activeSchema) {
                        setActiveSchema(nextSchema);
                      }
                    }

                    lastActiveTabIndex = activeTabIndex;
                    lastTabAmount = tabs.length;
                  },
                  toolbar: {
                    additionalComponent: function () {
                      const schema = schemas[activeSchema];

                      return React.createElement(
                        GraphiQL.React.ToolbarButton,
                        {
                          onClick: () => {
                            const activeKeyIndex = keys.indexOf(activeSchema);
                            nextSchemaKey =
                              keys[(activeKeyIndex + 1) % keys.length];

                            // This triggers onTabChange
                            if (nextSchemaKey) setActiveSchema(nextSchemaKey);
                          },
                          label: 'Toggle between different API schemas',
                        },
                        React.createElement(
                          'div',
                          {
                            key: 'api-wrapper',
                            className: 'graphiql-toolbar-icon',
                            style: {position: 'relative', fontWeight: 'bolder'},
                          },
                          [
                            React.createElement(
                              'div',
                              {key: 'icon', style: {textAlign: 'center'}},
                              [
                                schema.icon,
                                React.createElement(
                                  'div',
                                  {
                                    key: 'icon-label',
                                    className: 'graphiql-api-toolbar-label',
                                  },
                                  'API',
                                ),
                              ],
                            ),
                          ],
                        ),
                      );
                    },
                  },
                },
                [
                  React.createElement(
                    GraphiQL.Logo,
                    {
                      key: 'Logo replacement',
                    },
                    [
                      React.createElement(
                        'div',
                        {
                          key: 'Logo wrapper',
                          style: {display: 'flex', alignItems: 'center'},
                        },
                        [
                          React.createElement(
                            'div',
                            {
                              key: 'api',
                              className: 'graphiql-logo',
                              style: {
                                paddingRight: 0,
                                whiteSpace: 'nowrap',
                              },
                            },
                            [schema.name],
                          ),
                          React.createElement(GraphiQL.Logo, {key: 'logo'}),
                        ],
                      ),
                    ],
                  ),
                ],
              );
            }
          </script>
        </body>
      </html>
    `,{status:200,headers:{"content-type":"text/html"}})};var cn={"&":"\\u0026",">":"\\u003e","<":"\\u003c","\u2028":"\\u2028","\u2029":"\\u2029"},un=/[&><\u2028\u2029]/g;function xr(e){return e.replace(un,t=>cn[t])}var ne="Error in SEO input: ",Y={title:{validate:e=>{if(typeof e!="string")throw new Error(ne.concat("`title` should be a string"));if(typeof e=="string"&&e.length>120)throw new Error(ne.concat("`title` should not be longer than 120 characters"));return e}},description:{validate:e=>{if(typeof e!="string")throw new Error(ne.concat("`description` should be a string"));if(typeof e=="string"&&e.length>155)throw new Error(ne.concat("`description` should not be longer than 155 characters"));return e}},url:{validate:e=>{if(typeof e!="string")throw new Error(ne.concat("`url` should be a string"));if(typeof e=="string"&&!e.startsWith("http"))throw new Error(ne.concat("`url` should be a valid URL"));return e}},handle:{validate:e=>{if(typeof e!="string")throw new Error(ne.concat("`handle` should be a string"));if(typeof e=="string"&&!e.startsWith("@"))throw new Error(ne.concat("`handle` should start with `@`"));return e}}};function Dr(e){let t=[];for(let r of Object.keys(e))switch(r){case"title":{let o=z(Y.title,e.title),n=_t(e?.titleTemplate,o);if(!n)break;t.push(N("title",{title:n}),N("meta",{property:"og:title",content:n}),N("meta",{name:"twitter:title",content:n}));break}case"description":{let o=z(Y.description,e.description);if(!o)break;t.push(N("meta",{name:"description",content:o}),N("meta",{property:"og:description",content:o}),N("meta",{name:"twitter:description",content:o}));break}case"url":{let o=z(Y.url,e.url);if(!o)break;let a=o.split("?")[0].replace(/\/$/,"");t.push(N("link",{rel:"canonical",href:a}),N("meta",{property:"og:url",content:a}));break}case"handle":{let o=z(Y.handle,e.handle);if(!o)break;t.push(N("meta",{name:"twitter:site",content:o}),N("meta",{name:"twitter:creator",content:o}));break}case"media":{let o,n=ae(e.media);for(let a of n)if(typeof a=="string"&&t.push(N("meta",{name:"og:image",content:a})),a&&typeof a=="object"){let s=a.type||"image",c=a?{url:a?.url,secure_url:a?.url,type:Ut(a.url),width:a?.width,height:a?.height,alt:a?.altText}:{};for(let i of Object.keys(c))c[i]&&(o=c[i],t.push(N("meta",{property:`og:${s}:${i}`,content:o},c.url)));}break}case"jsonLd":{let o=ae(e.jsonLd),n=0;for(let a of o){if(typeof a!="object")continue;let s=N("script",{type:"application/ld+json",children:JSON.stringify(a,(c,i)=>typeof i=="string"?xr(i):i)},`json-ld-${a?.["@type"]||a?.name||n++}`);t.push(s);}break}case"alternates":{let o=ae(e.alternates);for(let n of o){if(!n)continue;let{language:a,url:s,default:c}=n,i=a?`${a}${c?"-default":""}`:void 0;t.push(N("link",{rel:"alternate",hrefLang:i,href:s}));}break}case"robots":{if(!e.robots)break;let{maxImagePreview:o,maxSnippet:n,maxVideoPreview:a,noArchive:s,noFollow:c,noImageIndex:i,noIndex:u,noSnippet:l,noTranslate:y,unavailableAfter:g}=e.robots,d=[s&&"noarchive",i&&"noimageindex",l&&"nosnippet",y&&"notranslate",o&&`max-image-preview:${o}`,n&&`max-snippet:${n}`,a&&`max-video-preview:${a}`,g&&`unavailable_after:${g}`],p=(u?"noindex":"index")+","+(c?"nofollow":"follow");for(let f of d)f&&(p+=`,${f}`);t.push(N("meta",{name:"robots",content:p}));break}}return t.flat().sort((r,o)=>r.key.localeCompare(o.key))}function N(e,t,r){let o={tag:e,props:{},key:""};return e==="title"?(o.children=t.title,o.key=Dt(o),o):e==="script"?(o.children=typeof t.children=="string"?t.children:"",o.key=Dt(o,r),delete t.children,o.props=t,o):(o.props=t,Object.keys(o.props).forEach(n=>!o.props[n]&&delete o.props[n]),o.key=Dt(o,r),o)}function Dt(e,t){let{tag:r,props:o}=e;if(r==="title")return "0-title";if(r==="meta"){let n=o.content===t&&typeof o.property=="string"&&!o.property.endsWith("secure_url")&&"0";return [r,...[t,n],o.property||o.name].filter(s=>s).join("-")}return r==="link"?[r,o.rel,o.hrefLang||o.media].filter(a=>a).join("-").replace(/\s+/g,"-"):r==="script"?`${r}-${t}`:`${r}-${o.type}`}function _t(e,t){if(t)return e?typeof e=="function"?e(t):e.replace("%s",t??""):t}function Ut(e){switch(e&&e.split(".").pop()){case"svg":return "image/svg+xml";case"png":return "image/png";case"gif":return "image/gif";case"swf":return "application/x-shockwave-flash";case"mp3":return "audio/mpeg";case"jpg":case"jpeg":default:return "image/jpeg"}}function ae(e){return Array.isArray(e)?e:[e]}function z(e,t){try{return e.validate(t)}catch(r){return console.warn(r.message),t}}var Cn=react.lazy(()=>Promise.resolve().then(()=>(Lr(),kr)));function Sn({debug:e}){let t=react$1.useMatches(),r=react$1.useLocation();console.warn("[h2:warn:Seo] The `<Seo/>` component is deprecated. Use `getSeoMeta` instead.\nSee: https://shopify.dev/docs/api/hydrogen/2024-10/utilities/getseometa");let o=react.useMemo(()=>t.flatMap(s=>{let{handle:c,...i}=s,u={...i,...r},l=c?.seo,y=i?.data?.seo;return !l&&!y?[]:l?ze(l,u):[y]}).reduce((s,c)=>{Object.keys(c).forEach(u=>!c[u]&&delete c[u]);let{jsonLd:i}=c;return i?s?.jsonLd?Array.isArray(i)?{...s,...c,jsonLd:[...s.jsonLd,...i]}:{...s,...c,jsonLd:[...s.jsonLd,i]}:{...s,...c,jsonLd:[i]}:{...s,...c}},{}),[t,r]),{html:n,loggerMarkup:a}=react.useMemo(()=>{let s=Dr(o),c=s.map(u=>u.tag==="script"?react.createElement(u.tag,{...u.props,key:u.key,dangerouslySetInnerHTML:{__html:u.children}}):react.createElement(u.tag,{...u.props,key:u.key},u.children)),i=react.createElement(react.Suspense,{fallback:null},react.createElement(Cn,{headTags:s}));return {html:c,loggerMarkup:i}},[o]);return react.createElement(react.Fragment,null,n,e&&a)}function ze(e,...t){if(e instanceof Function)return ze(e(...t),...t);let r={};return Array.isArray(e)?(r=e.reduce((o,n)=>[...o,ze(n)],[]),r):e instanceof Object?(Object.entries(e).forEach(([n,a])=>{r[n]=ze(a,...t);}),r):e}function Pn(...e){let t=[],r=e.reduce((o,n)=>{if(!n)return o;Object.keys(n).forEach(s=>!n[s]&&delete n[s]);let{jsonLd:a}=n;return a?o?.jsonLd?{...o,...n,jsonLd:ae(o.jsonLd).concat(a)}:{...o,...n,jsonLd:[a]}:{...o,...n}},{})||{};for(let o of Object.keys(r))switch(o){case"title":{let n=z(Y.title,r.title),a=_t(r?.titleTemplate,n);if(!a)break;t.push({title:a},{property:"og:title",content:a},{property:"twitter:title",content:a});break}case"description":{let n=z(Y.description,r.description);if(!n)break;t.push({name:"description",content:n},{property:"og:description",content:n},{property:"twitter:description",content:n});break}case"url":{let n=z(Y.url,r.url);if(!n)break;let s=n.split("?")[0].replace(/\/$/,"");t.push({tagName:"link",rel:"canonical",href:s},{property:"og:url",content:s});break}case"handle":{let n=z(Y.handle,r.handle);if(!n)break;t.push({property:"twitter:site",content:n},{property:"twitter:creator",content:n});break}case"media":{let n,a=ae(r.media);for(let s of a)if(typeof s=="string"&&t.push({property:"og:image",content:s}),s&&typeof s=="object"){let c=s.type||"image",i=s?{url:s?.url,secure_url:s?.url,type:Ut(s.url),width:s?.width,height:s?.height,alt:s?.altText}:{};for(let u of Object.keys(i))i[u]&&(n=i[u],t.push({property:`og:${c}:${u}`,content:n}));}break}case"jsonLd":{let n=ae(r.jsonLd);for(let s of n)typeof s!="object"||Object.keys(s).length===0||t.push({"script:ld+json":s});break}case"alternates":{let n=ae(r.alternates);for(let a of n){if(!a)continue;let{language:s,url:c,default:i}=a,u=s?`${s}${i?"-default":""}`:void 0;t.push({tagName:"link",rel:"alternate",hrefLang:u,href:c});}break}case"robots":{if(!r.robots)break;let{maxImagePreview:n,maxSnippet:a,maxVideoPreview:s,noArchive:c,noFollow:i,noImageIndex:u,noIndex:l,noSnippet:y,noTranslate:g,unavailableAfter:d}=r.robots,p=[c&&"noarchive",u&&"noimageindex",y&&"nosnippet",g&&"notranslate",n&&`max-image-preview:${n}`,a&&`max-snippet:${a}`,s&&`max-video-preview:${s}`,d&&`unavailable_after:${d}`],f=(l?"noindex":"index")+","+(i?"nofollow":"follow");for(let P of p)P&&(f+=`,${P}`);t.push({name:"robots",content:f});break}}return t}function vn({connection:e,children:t=()=>(console.warn("<Pagination> requires children to work properly"),null),namespace:r=""}){let [o,n]=react.useState(!1),a=react$1.useNavigation(),s=react$1.useLocation();react$1.useNavigate();react.useEffect(()=>{a.state==="idle"&&n(!1);},[a.state]);let{endCursor:i,hasNextPage:u,hasPreviousPage:l,nextPageUrl:y,nodes:g,previousPageUrl:d,startCursor:p}=In(e,r),f=react.useMemo(()=>({...s.state,pagination:{...s.state?.pagination||{},[r]:{pageInfo:{endCursor:i,hasPreviousPage:l,hasNextPage:u,startCursor:p},nodes:g}}}),[i,u,l,p,g,r,s.state]),P=react.useMemo(()=>react.forwardRef(function(A,T){return u?react.createElement(react$1.Link,{preventScrollReset:!0,...A,to:y,state:f,replace:!0,ref:T,onClick:()=>n(!0)}):null}),[u,y,f]),m=react.useMemo(()=>react.forwardRef(function(A,T){return l?react.createElement(react$1.Link,{preventScrollReset:!0,...A,to:d,state:f,replace:!0,ref:T,onClick:()=>n(!0)}):null}),[l,d,f]);return t({state:f,hasNextPage:u,hasPreviousPage:l,isLoading:o,nextPageUrl:y,nodes:g,previousPageUrl:d,NextLink:P,PreviousLink:m})}function Je(e,t){let r=new URLSearchParams(e);return Object.keys(t?.pagination||{}).forEach(n=>{let a=n===""?"":`${n}_`,s=`${a}cursor`,c=`${a}direction`;r.delete(s),r.delete(c);}),r.toString()}function Re(e){throw new Error(`The Pagination component requires ${"`"+e+"`"} to be a part of your query. See the guide on how to setup your query to include ${"`"+e+"`"}: https://shopify.dev/docs/custom-storefronts/hydrogen/data-fetching/pagination#setup-the-paginated-query`)}function In(e,t=""){e.pageInfo||Re("pageInfo"),typeof e.pageInfo.startCursor>"u"&&Re("pageInfo.startCursor"),typeof e.pageInfo.endCursor>"u"&&Re("pageInfo.endCursor"),typeof e.pageInfo.hasNextPage>"u"&&Re("pageInfo.hasNextPage"),typeof e.pageInfo.hasPreviousPage>"u"&&Re("pageInfo.hasPreviousPage");let r=react$1.useNavigation(),o=react$1.useNavigate(),{state:n,search:a,pathname:s}=react$1.useLocation(),c=t?`${t}_cursor`:"cursor",i=t?`${t}_direction`:"direction",y=new URLSearchParams(a).get(i)==="previous",g=react.useMemo(()=>!globalThis?.window?.__hydrogenHydrated||!n?.pagination?.[t]?.nodes?hydrogenReact.flattenConnection(e):y?[...hydrogenReact.flattenConnection(e),...n.pagination[t].nodes||[]]:[...n.pagination[t].nodes||[],...hydrogenReact.flattenConnection(e)],[n,e,t]),d=react.useMemo(()=>{let m=globalThis?.window?.__hydrogenHydrated,C=n?.pagination?.[t]?.pageInfo,A=!m||C?.startCursor===void 0?e.pageInfo.startCursor:C.startCursor,T=!m||C?.endCursor===void 0?e.pageInfo.endCursor:C.endCursor,h=!m||C?.hasPreviousPage===void 0?e.pageInfo.hasPreviousPage:C.hasPreviousPage,I=!m||C?.hasNextPage===void 0?e.pageInfo.hasNextPage:C.hasNextPage;return n?.pagination?.[t]?.nodes&&(y?(A=e.pageInfo.startCursor,h=e.pageInfo.hasPreviousPage):(T=e.pageInfo.endCursor,I=e.pageInfo.hasNextPage)),{startCursor:A,endCursor:T,hasPreviousPage:h,hasNextPage:I}},[y,n,t,e.pageInfo.hasNextPage,e.pageInfo.hasPreviousPage,e.pageInfo.startCursor,e.pageInfo.endCursor]),p=react.useRef({params:Je(a,n),pathname:s});react.useEffect(()=>{window.__hydrogenHydrated=!0;},[]),react.useEffect(()=>{let m=Je(a,n),C=p.current.params;(s!==p.current.pathname||m!==C)&&!(r.state==="idle"&&!r.location)&&(p.current={pathname:s,params:Je(a,n)},o(`${s}?${Je(a,n)}`,{replace:!0,preventScrollReset:!0,state:{nodes:void 0,pageInfo:void 0}}));},[s,a,n]);let f=react.useMemo(()=>{let m=new URLSearchParams(a);return m.set(i,"previous"),d.startCursor&&m.set(c,d.startCursor),`?${m.toString()}`},[a,d.startCursor]),P=react.useMemo(()=>{let m=new URLSearchParams(a);return m.set(i,"next"),d.endCursor&&m.set(c,d.endCursor),`?${m.toString()}`},[a,d.endCursor]);return {...d,previousPageUrl:f,nextPageUrl:P,nodes:g}}function En(e,t={pageBy:20}){if(typeof e?.url>"u")throw new Error("getPaginationVariables must be called with the Request object passed to your loader function");let{pageBy:r,namespace:o=""}=t,n=new URLSearchParams(new URL(e.url).search),a=o?`${o}_cursor`:"cursor",s=o?`${o}_direction`:"direction",c=n.get(a)??void 0;return (n.get(s)==="previous"?"previous":"next")==="previous"?{last:r,startCursor:c??null}:{first:r,endCursor:c??null}}var Xe="2024-10",me=`Shopify Hydrogen ${Te}`,Br="30243aa5-17c1-465a-8493-944bcc4e88aa",V="customerAccount",ge="buyer";var q=class extends Response{constructor(t,r,o){super(`Bad request: ${t}`,{status:400,headers:o});}};function we(e,t={}){let r=t.headers?new Headers(t.headers):new Headers({});return r.set("location",e),new Response(null,{status:t.status||302,headers:r})}async function Rn({session:e,customerAccountId:t,customerAccountTokenExchangeUrl:r,httpsOrigin:o,debugInfo:n,exchangeForStorefrontCustomerAccessToken:a}){let s=new URLSearchParams,c=e.get(V),i=c?.refreshToken,u=c?.idToken;if(!i)throw new q("Unauthorized","No refreshToken found in the session. Make sure your session is configured correctly and passed to `createCustomerAccountClient`.");s.append("grant_type","refresh_token"),s.append("refresh_token",i),s.append("client_id",t);let l={"content-type":"application/x-www-form-urlencoded","User-Agent":me,Origin:o};new Date().getTime();let g=r,d=await fetch(g,{method:"POST",headers:l,body:s});if(!d.ok){let C=await d.text();throw new Response(C,{status:d.status,headers:{"Content-Type":"text/html; charset=utf-8"}})}let{access_token:p,expires_in:f,refresh_token:P}=await d.json(),m=await Vt(p,t,r,o);e.set(V,{accessToken:m,expiresAt:new Date(new Date().getTime()+(f-120)*1e3).getTime()+"",refreshToken:P,idToken:u}),await a();}function he(e){e.unset(V),e.unset(ge);}async function Gr({locks:e,expiresAt:t,session:r,customerAccountId:o,customerAccountTokenExchangeUrl:n,httpsOrigin:a,debugInfo:s,exchangeForStorefrontCustomerAccessToken:c}){if(parseInt(t,10)-1e3<new Date().getTime())try{e.refresh||(e.refresh=Rn({session:r,customerAccountId:o,customerAccountTokenExchangeUrl:n,httpsOrigin:a,debugInfo:s,exchangeForStorefrontCustomerAccessToken:c})),await e.refresh,delete e.refresh;}catch(i){throw he(r),i&&i.status!==401?i:new q("Unauthorized","Login before querying the Customer Account API.")}}function Qr(){let e=bn();return jr(e)}async function Wr(e){let t=await crypto.subtle.digest({name:"SHA-256"},new TextEncoder().encode(e)),r=wn(t);return jr(r)}function bn(){let e=new Uint8Array(32);return crypto.getRandomValues(e),String.fromCharCode.apply(null,Array.from(e))}function jr(e){return btoa(e).replace(/\+/g,"-").replace(/\//g,"_").replace(/=/g,"")}function wn(e){let t=new Uint8Array(e),r=Array.from(t);return String.fromCharCode(...r)}function Kr(){let e=Date.now().toString(),t=Math.random().toString(36).substring(2);return e+t}async function Vt(e,t,r,o,n){let a=t;if(!e)throw new q("Unauthorized","oAuth access token was not provided during token exchange.");let s=new URLSearchParams;s.append("grant_type","urn:ietf:params:oauth:grant-type:token-exchange"),s.append("client_id",a),s.append("audience",Br),s.append("subject_token",e),s.append("subject_token_type","urn:ietf:params:oauth:token-type:access_token"),s.append("scopes","https://api.customers.com/auth/customer.graphql");let c={"content-type":"application/x-www-form-urlencoded","User-Agent":me,Origin:o};new Date().getTime();let u=r,l=await fetch(u,{method:"POST",headers:c,body:s});let y=await l.json();if(y.error)throw new q(y.error_description);return y.access_token}function Yr(e){return On(e).payload.nonce}function On(e){let[t,r,o]=e.split("."),n=JSON.parse(atob(t)),a=JSON.parse(atob(r));return {header:n,payload:a,signature:o}}function Ze(){return Dn(xn())}function xn(){try{return crypto.getRandomValues(new Uint8Array(16))}catch{return new Uint8Array(16).map(()=>Math.random()*255|0)}}function Dn(e){return Array.from(e,function(t){return ("0"+(t&255).toString(16)).slice(-2)}).join("")}function zr(e,t,r){let o=r?`https://shopify.com/${r}`:t,n=r?`https://shopify.com/authentication/${r}`:`${t}/auth`;return function(s){switch(s){case"CA_BASE_URL":return o;case"CA_BASE_AUTH_URL":return n;case"GRAPHQL":return `${o}/account/customer/api/${e}/graphql`;case"AUTH":return `${n}/oauth/authorize`;case"LOGIN_SCOPE":return r?"openid email customer-account-api:full":"openid email https://api.customers.com/auth/customer.graphql";case"TOKEN_EXCHANGE":return `${n}/oauth/token`;case"LOGOUT":return `${n}/logout`}}}function _n(e,t){if(!e.url)return t;let{pathname:r}=new URL(e.url),o=t+`?${new URLSearchParams({return_to:r}).toString()}`;return we(o)}function Mt({session:e,customerAccountId:t,customerAccountUrl:r,shopId:o,customerApiVersion:n=Xe,request:a,waitUntil:s,authUrl:c,customAuthStatusHandler:i,logErrors:u=!0,unstableB2b:l=!1,loginPath:y="/account/login",authorizePath:g="/account/authorize",defaultRedirectPath:d="/account"}){if(n!==Xe&&console.warn(`[h2:warn:createCustomerAccountClient] You are using Customer Account API version ${n} when this version of Hydrogen was built for ${Xe}.`),e||console.warn("[h2:warn:createCustomerAccountClient] session is required to use Customer Account API. Ensure the session object passed in exist."),!a?.url)throw new Error("[h2:error:createCustomerAccountClient] The request object does not contain a URL.");r&&!o&&re("[h2:warn:createCustomerAccountClient] The `customerAccountUrl` option is deprecated and will be removed in a future version. Please remove `customerAccountUrl` and supply a `shopId: env.SHOP_ID` option instead.\n\nIf using `createHydrogenContext`, ensure there is a SHOP_ID defined in your local .env file.");let p=i||(()=>_n(a,y)),f=new URL(a.url),P=f.protocol==="http:"?f.origin.replace("http","https"):f.origin,m=wt({requestUrl:P,defaultUrl:g,redirectUrl:c}),C=zr(n,r,o),A=Un(C,t),T=C("GRAPHQL"),h={};async function I({query:v,type:E,variables:b={}}){let w=await U();if(!w)throw p();new Date().getTime();let _=await fetch(T,{method:"POST",headers:{"Content-Type":"application/json","User-Agent":me,Origin:P,Authorization:w},body:JSON.stringify({query:v,variables:b})});let F=await _.text(),$={url:T,response:_,type:E,query:v,queryVariables:b,errors:void 0,client:"customer"};if(!_.ok){if(_.status===401)throw he(e),p();let k;try{k=K(F);}catch{k=[{message:F}];}ve({...$,errors:k});}try{let k=K(F),{errors:j}=k,L=j?.map(({message:G,...Ne})=>new ue(G,{...Ne,clientOperation:`customerAccount.${$.type}`,requestId:_.headers.get("x-request-id"),queryVariables:b,query:v}));return {...k,...j&&{errors:L}}}catch{ve({...$,errors:[{message:F}]});}}async function S(){if(!o&&(!r||!t))return !1;let v=e.get(V),E=v?.accessToken,b=v?.expiresAt;if(!E||!b)return !1;let w=Q?.();try{await Gr({locks:h,expiresAt:b,session:e,customerAccountId:t,customerAccountTokenExchangeUrl:C("TOKEN_EXCHANGE"),httpsOrigin:P,debugInfo:{waitUntil:s,stackInfo:w,...oe(a)},exchangeForStorefrontCustomerAccessToken:ee});}catch{return !1}return !0}async function M(){if(!await S())throw p()}async function U(){if(await S())return e.get(V)?.accessToken}async function Z(v,E){return A(),v=ye(v),Ke(v,"customer.mutate"),fe(I({query:v,type:"mutation",...E}),{logErrors:u})}async function ct(v,E){return A(),v=ye(v),je(v,"customer.query"),fe(I({query:v,type:"query",...E}),{logErrors:u})}function Le(v){e.set(ge,{...e.get(ge),...v});}async function ut(){if(await S())return e.get(ge)}async function ee(){if(!l)return;let v=`#graphql
      mutation storefrontCustomerAccessTokenCreate {
        storefrontCustomerAccessTokenCreate {
          customerAccessToken
        }
      }
    `,{data:E}=await Z(v),b=E?.storefrontCustomerAccessTokenCreate?.customerAccessToken;b&&Le({customerAccessToken:b});}return {login:async v=>{A();let E=new URL(C("AUTH")),b=Kr(),w=Ze();if(E.searchParams.set("client_id",t),E.searchParams.set("scope","openid email"),E.searchParams.append("response_type","code"),E.searchParams.append("redirect_uri",m),E.searchParams.set("scope",C("LOGIN_SCOPE")),E.searchParams.append("state",b),E.searchParams.append("nonce",w),v?.uiLocales){let[_,F]=v.uiLocales.split("-"),$=_.toLowerCase();F&&($+=`-${F.toUpperCase()}`),E.searchParams.append("ui_locales",$);}let W=Qr(),te=await Wr(W);return e.set(V,{...e.get(V),codeVerifier:W,state:b,nonce:w,redirectPath:Ye(a.url)||B(a,"Referer")||d}),E.searchParams.append("code_challenge",te),E.searchParams.append("code_challenge_method","S256"),we(E.toString())},logout:async v=>{A();let E=e.get(V)?.idToken,b=wt({requestUrl:P,defaultUrl:P,redirectUrl:v?.postLogoutRedirectUri}),w=E?new URL(`${C("LOGOUT")}?${new URLSearchParams([["id_token_hint",E],["post_logout_redirect_uri",b]]).toString()}`).toString():b;return he(e),we(w,{headers:v?.headers||{}})},isLoggedIn:S,handleAuthStatus:M,getAccessToken:U,getApiUrl:()=>T,mutate:Z,query:ct,authorize:async()=>{A();let v=f.searchParams.get("code"),E=f.searchParams.get("state");if(!v||!E)throw he(e),new q("Unauthorized","No code or state parameter found in the redirect URL.");if(e.get(V)?.state!==E)throw he(e),new q("Unauthorized","The session state does not match the state parameter. Make sure that the session is configured correctly and passed to `createCustomerAccountClient`.");let b=t,w=new URLSearchParams;w.append("grant_type","authorization_code"),w.append("client_id",b),w.append("redirect_uri",m),w.append("code",v);let W=e.get(V)?.codeVerifier;if(!W)throw new q("Unauthorized","No code verifier found in the session. Make sure that the session is configured correctly and passed to `createCustomerAccountClient`.");w.append("code_verifier",W);let te={"content-type":"application/x-www-form-urlencoded","User-Agent":me,Origin:P},_=Q?.();new Date().getTime();let $=C("TOKEN_EXCHANGE"),k=await fetch($,{method:"POST",headers:te,body:w});if(!k.ok)throw new Response(await k.text(),{status:k.status,headers:{"Content-Type":"text/html; charset=utf-8"}});let{access_token:j,expires_in:L,id_token:G,refresh_token:Ne}=await k.json(),cr=e.get(V)?.nonce,ur=await Yr(G);if(cr!==ur)throw new q("Unauthorized",`Returned nonce does not match: ${cr} !== ${ur}`);let pr=j;o||(pr=await Vt(j,t,C("TOKEN_EXCHANGE"),P,{waitUntil:s,stackInfo:_,...oe(a)}));let ko=e.get(V)?.redirectPath;return e.set(V,{accessToken:pr,expiresAt:new Date(new Date().getTime()+(L-120)*1e3).getTime()+"",refreshToken:Ne,idToken:G}),await ee(),we(ko||d)},UNSTABLE_setBuyer:Le,UNSTABLE_getBuyer:ut}}function Un(e,t){return function(){try{if(!t)throw Error();new URL(e("CA_BASE_URL")),new URL(e("CA_BASE_AUTH_URL"));}catch{console.error(new Error("[h2:error:customerAccount] You do not have the valid credential to use Customer Account API.\nRun `h2 env pull` to link your store credentials."));let o="Internal Server Error";throw new Response(o,{status:500})}}}var kn="https://raw.githubusercontent.com/Shopify/hydrogen/main/docs/changelog.json";async function Ln({request:e,changelogUrl:t}){new URL(e.url).searchParams;return fetch(t||kn)}var Jr="cartFormInput";function J({children:e,action:t,inputs:r,route:o,fetcherKey:n}){let a=react$1.useFetcher({key:n});return jsxRuntime.jsxs(a.Form,{action:o||"",method:"post",children:[(t||r)&&jsxRuntime.jsx("input",{type:"hidden",name:Jr,value:JSON.stringify({action:t,inputs:r})}),typeof e=="function"?e(a):e]})}J.INPUT_NAME=Jr;J.ACTIONS={AttributesUpdateInput:"AttributesUpdateInput",BuyerIdentityUpdate:"BuyerIdentityUpdate",Create:"Create",DiscountCodesUpdate:"DiscountCodesUpdate",GiftCardCodesUpdate:"GiftCardCodesUpdate",LinesAdd:"LinesAdd",LinesRemove:"LinesRemove",LinesUpdate:"LinesUpdate",NoteUpdate:"NoteUpdate",SelectedDeliveryOptionsUpdate:"SelectedDeliveryOptionsUpdate",MetafieldsSet:"MetafieldsSet",MetafieldDelete:"MetafieldDelete"};function Vn(e){let t={};for(let s of e.entries()){let c=s[0],i=e.getAll(c);t[c]=i.length>1?i:s[1];}let{cartFormInput:r,...o}=t,{action:n,inputs:a}=r?JSON.parse(String(r)):{};return {action:n,inputs:{...a,...o}}}J.getFormInput=Vn;var O=`#graphql
  fragment CartApiError on CartUserError {
    message
    field
    code
  }
`,x=`#graphql
  fragment CartApiMutation on Cart {
    id
    totalQuantity
    checkoutUrl
  }
`,D=`#graphql
  fragment CartApiWarning on CartWarning {
    code
    message
    target
  }
`;function Ht(e){return async(t,r)=>{let o=e.customerAccount?await e.customerAccount.UNSTABLE_getBuyer():void 0,{cartId:n,...a}=r||{},{buyerIdentity:s,...c}=t,{cartCreate:i,errors:u}=await e.storefront.mutate(Fn(e.cartFragment),{variables:{input:{...c,buyerIdentity:{...o,...s}},...a}});return R(i,u)}}var Fn=(e=x)=>`#graphql
  mutation cartCreate(
    $input: CartInput!
    $country: CountryCode = ZZ
    $language: LanguageCode
  ) @inContext(country: $country, language: $language) {
    cartCreate(input: $input) {
      cart {
        ...CartApiMutation
        checkoutUrl
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${O}
  ${D}
`;function Ft({storefront:e,customerAccount:t,getCartId:r,cartFragment:o}){return async n=>{let a=r();if(!a)return null;let[s,{cart:c,errors:i}]=await Promise.all([t?t.isLoggedIn():!1,e.query(qn(o),{variables:{cartId:a,...n},cache:e.CacheNone()})]);if(s&&c?.checkoutUrl){let u=new URL(c.checkoutUrl);u.searchParams.set("logged_in","true"),c.checkoutUrl=u.toString();}return c||i?R(c,i):null}}var qn=(e=$n)=>`#graphql
  query CartQuery(
    $cartId: ID!
    $numCartLines: Int = 100
    $country: CountryCode = ZZ
    $language: LanguageCode
  ) @inContext(country: $country, language: $language) {
    cart(id: $cartId) {
      ...CartApiQuery
    }
  }

  ${e}
`,$n=`#graphql
  fragment CartApiQuery on Cart {
    updatedAt
    id
    checkoutUrl
    totalQuantity
    buyerIdentity {
      countryCode
      customer {
        id
        email
        firstName
        lastName
        displayName
      }
      email
      phone
    }
    lines(first: $numCartLines) {
      edges {
        node {
          id
          quantity
          attributes {
            key
            value
          }
          cost {
            totalAmount {
              amount
              currencyCode
            }
            amountPerQuantity {
              amount
              currencyCode
            }
            compareAtAmountPerQuantity {
              amount
              currencyCode
            }
          }
          merchandise {
            ... on ProductVariant {
              id
              availableForSale
              compareAtPrice {
                ...CartApiMoney
              }
              price {
                ...CartApiMoney
              }
              requiresShipping
              title
              image {
                ...CartApiImage
              }
              product {
                handle
                title
                id
                vendor
              }
              selectedOptions {
                name
                value
              }
            }
          }
        }
      }
    }
    cost {
      subtotalAmount {
        ...CartApiMoney
      }
      totalAmount {
        ...CartApiMoney
      }
      totalDutyAmount {
        ...CartApiMoney
      }
      totalTaxAmount {
        ...CartApiMoney
      }
    }
    note
    attributes {
      key
      value
    }
    discountCodes {
      applicable
      code
    }
  }

  fragment CartApiMoney on MoneyV2 {
    currencyCode
    amount
  }

  fragment CartApiImage on Image {
    id
    url
    altText
    width
    height
  }
`;function qt(e){return async(t,r)=>{let{cartLinesAdd:o,errors:n}=await e.storefront.mutate(Bn(e.cartFragment),{variables:{cartId:e.getCartId(),lines:t,...r}});return R(o,n)}}var Bn=(e=x)=>`#graphql
  mutation cartLinesAdd(
    $cartId: ID!
    $lines: [CartLineInput!]!
    $country: CountryCode = ZZ
    $language: LanguageCode
  ) @inContext(country: $country, language: $language) {
    cartLinesAdd(cartId: $cartId, lines: $lines) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${O}
  ${D}
`;var Xr="__h_pending_";function Zr(e){return Xr+e}function et(e){return e.startsWith(Xr)}function tt(e,t){if(t.some(r=>et(typeof r=="string"?r:r.id)))throw new Error(`Tried to perform an action on an optimistic line. Make sure to disable your "${e}" CartForm action when the line is optimistic.`)}function $t(e){return async(t,r)=>{tt("updateLines",t);let{cartLinesUpdate:o,errors:n}=await e.storefront.mutate(Gn(e.cartFragment),{variables:{cartId:e.getCartId(),lines:t,...r}});return R(o,n)}}var Gn=(e=x)=>`#graphql
  mutation cartLinesUpdate(
    $cartId: ID!
    $lines: [CartLineUpdateInput!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartLinesUpdate(cartId: $cartId, lines: $lines) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${O}
  ${D}
`;function Bt(e){return async(t,r)=>{tt("removeLines",t);let{cartLinesRemove:o,errors:n}=await e.storefront.mutate(Qn(e.cartFragment),{variables:{cartId:e.getCartId(),lineIds:t,...r}});return R(o,n)}}var Qn=(e=x)=>`#graphql
  mutation cartLinesRemove(
    $cartId: ID!
    $lineIds: [ID!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartLinesRemove(cartId: $cartId, lineIds: $lineIds) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${O}
  ${D}
`;function Gt(e){return async(t,r)=>{let o=t.filter((s,c,i)=>i.indexOf(s)===c),{cartDiscountCodesUpdate:n,errors:a}=await e.storefront.mutate(Wn(e.cartFragment),{variables:{cartId:e.getCartId(),discountCodes:o,...r}});return R(n,a)}}var Wn=(e=x)=>`#graphql
  mutation cartDiscountCodesUpdate(
    $cartId: ID!
    $discountCodes: [String!]
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartDiscountCodesUpdate(cartId: $cartId, discountCodes: $discountCodes) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${O}
  ${D}
`;function Qt(e){return async(t,r)=>{let o=t.filter((s,c,i)=>i.indexOf(s)===c),{cartGiftCardCodesUpdate:n,errors:a}=await e.storefront.mutate(jn(e.cartFragment),{variables:{cartId:e.getCartId(),giftCardCodes:o,...r}});return R(n,a)}}var jn=(e=x)=>`#graphql
  mutation cartGiftCardCodesUpdate(
    $cartId: ID!
    $giftCardCodes: [String!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartGiftCardCodesUpdate(cartId: $cartId, giftCardCodes: $giftCardCodes) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${O}
  ${D}
`;function Wt(e){return async(t,r)=>{t.companyLocationId&&e.customerAccount&&e.customerAccount.UNSTABLE_setBuyer({companyLocationId:t.companyLocationId});let o=e.customerAccount?await e.customerAccount.UNSTABLE_getBuyer():void 0,{cartBuyerIdentityUpdate:n,errors:a}=await e.storefront.mutate(Kn(e.cartFragment),{variables:{cartId:e.getCartId(),buyerIdentity:{...o,...t},...r}});return R(n,a)}}var Kn=(e=x)=>`#graphql
  mutation cartBuyerIdentityUpdate(
    $cartId: ID!
    $buyerIdentity: CartBuyerIdentityInput!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartBuyerIdentityUpdate(cartId: $cartId, buyerIdentity: $buyerIdentity) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${O}
  ${D}
`;function jt(e){return async(t,r)=>{let{cartNoteUpdate:o,errors:n}=await e.storefront.mutate(Yn(e.cartFragment),{variables:{cartId:e.getCartId(),note:t,...r}});return R(o,n)}}var Yn=(e=x)=>`#graphql
  mutation cartNoteUpdate(
    $cartId: ID!
    $note: String!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartNoteUpdate(cartId: $cartId, note: $note) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${O}
  ${D}
`;function Kt(e){return async(t,r)=>{let{cartSelectedDeliveryOptionsUpdate:o,errors:n}=await e.storefront.mutate(zn(e.cartFragment),{variables:{cartId:e.getCartId(),selectedDeliveryOptions:t,...r}});return R(o,n)}}var zn=(e=x)=>`#graphql
  mutation cartSelectedDeliveryOptionsUpdate(
    $cartId: ID!
    $selectedDeliveryOptions: [CartSelectedDeliveryOptionInput!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartSelectedDeliveryOptionsUpdate(cartId: $cartId, selectedDeliveryOptions: $selectedDeliveryOptions) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${O}
  ${D}
`;function Yt(e){return async(t,r)=>{let{cartAttributesUpdate:o,errors:n}=await e.storefront.mutate(Jn(e.cartFragment),{variables:{cartId:r?.cartId||e.getCartId(),attributes:t}});return R(o,n)}}var Jn=(e=x)=>`#graphql
  mutation cartAttributesUpdate(
    $cartId: ID!
    $attributes: [AttributeInput!]!
  ) {
    cartAttributesUpdate(cartId: $cartId, attributes: $attributes) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${O}
  ${D}
`;function zt(e){return async(t,r)=>{let o=r?.cartId||e.getCartId(),n=t.map(c=>({...c,ownerId:o})),{cartMetafieldsSet:a,errors:s}=await e.storefront.mutate(Xn(),{variables:{metafields:n}});return R({cart:{id:o},...a},s)}}var Xn=()=>`#graphql
  mutation cartMetafieldsSet(
    $metafields: [CartMetafieldsSetInput!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartMetafieldsSet(metafields: $metafields) {
      userErrors {
        code
        elementIndex
        field
        message
      }
    }
  }
`;function Jt(e){return async(t,r)=>{let o=r?.cartId||e.getCartId(),{cartMetafieldDelete:n,errors:a}=await e.storefront.mutate(Zn(),{variables:{input:{ownerId:o,key:t}}});return R({cart:{id:o},...n},a)}}var Zn=()=>`#graphql
  mutation cartMetafieldDelete(
    $input: CartMetafieldDeleteInput!
  ) {
    cartMetafieldDelete(input: $input) {
      userErrors {
        code
        field
        message
      }
    }
  }
`;var Xt=e=>{let t=cookie.parse(vt(e,"Cookie")||"");return ()=>t.cart?`gid://shopify/Cart/${t.cart}`:void 0};var Zt=e=>t=>{let r=new Headers;return r.append("Set-Cookie",cookie.stringify("cart",t.split("/").pop()||"",{path:"/",...e})),r};function er(e){let{getCartId:t,setCartId:r,storefront:o,customerAccount:n,cartQueryFragment:a,cartMutateFragment:s}=e,c=t(),i=()=>c||t(),u={storefront:o,getCartId:i,cartFragment:s,customerAccount:n},l=Ht(u),y=async function(...d){let p=await l(...d);return c=p?.cart?.id,p},g={get:Ft({storefront:o,customerAccount:n,getCartId:i,cartFragment:a}),getCartId:i,setCartId:r,create:y,addLines:async(d,p)=>{let f=d.map(P=>({attributes:P.attributes,quantity:P.quantity,merchandiseId:P.merchandiseId,sellingPlanId:P.sellingPlanId}));return c||p?.cartId?await qt(u)(f,p):await y({lines:f},p)},updateLines:$t(u),removeLines:Bt(u),updateDiscountCodes:async(d,p)=>c||p?.cartId?await Gt(u)(d,p):await y({discountCodes:d},p),updateGiftCardCodes:async(d,p)=>c||p?.cartId?await Qt(u)(d,p):await y({giftCardCodes:d},p),updateBuyerIdentity:async(d,p)=>c||p?.cartId?await Wt(u)(d,p):await y({buyerIdentity:d},p),updateNote:async(d,p)=>c||p?.cartId?await jt(u)(d,p):await y({note:d},p),updateSelectedDeliveryOption:Kt(u),updateAttributes:async(d,p)=>c||p?.cartId?await Yt(u)(d,p):await y({attributes:d},p),setMetafields:async(d,p)=>c||p?.cartId?await zt(u)(d,p):await y({metafields:d},p),deleteMetafield:Jt(u)};return "customMethods"in e?{...g,...e.customMethods??{}}:g}function oa(e){let t=react$1.useFetchers();if(!t||!t.length)return e;let r=e?.lines?structuredClone(e):{lines:{nodes:[]}},o=r.lines.nodes,n=!1;for(let{formData:a}of t){if(!a)continue;let s=J.getFormInput(a);if(s.action===J.ACTIONS.LinesAdd)for(let c of s.inputs.lines){if(!c.selectedVariant){console.error("[h2:error:useOptimisticCart] No selected variant was passed in the cart action. Make sure to pass the selected variant if you want to use an optimistic cart");continue}let i=o.find(u=>u.merchandise.id===c.selectedVariant?.id);n=!0,i?(i.quantity=(i.quantity||1)+(c.quantity||1),i.isOptimistic=!0):o.unshift({id:Zr(c.selectedVariant.id),merchandise:c.selectedVariant,isOptimistic:!0,quantity:c.quantity||1});}else if(s.action===J.ACTIONS.LinesRemove)for(let c of s.inputs.lineIds){let i=o.findIndex(u=>u.id===c);if(i!==-1){if(et(o[i].id)){console.error("[h2:error:useOptimisticCart] Tried to remove an optimistic line that has not been added to the cart yet");continue}o.splice(i,1),n=!0;}else console.warn(`[h2:warn:useOptimisticCart] Tried to remove line '${c}' but it doesn't exist in the cart`);}else if(s.action===J.ACTIONS.LinesUpdate)for(let c of s.inputs.lines){let i=o.findIndex(u=>c.id===u.id);if(i>-1){if(et(o[i].id)){console.error("[h2:error:useOptimisticCart] Tried to update an optimistic line that has not been added to the cart yet");continue}o[i].quantity=c.quantity,o[i].quantity===0&&o.splice(i,1),n=!0;}else console.warn(`[h2:warn:useOptimisticCart] Tried to update line '${c.id}' but it doesn't exist in the cart`);}}return n&&(r.isOptimistic=n),r.totalQuantity=o.reduce((a,s)=>a+s.quantity,0),r}function ua({handle:e,options:t=[],variants:r=[],productPath:o="products",waitForNavigation:n=!1,selectedVariant:a,children:s}){let c=t;c[0]?.values&&(re("[h2:warn:VariantSelector] product.options.values is deprecated. Use product.options.optionValues instead."),c[0]&&!c[0].optionValues&&(c=t.map(p=>({...p,optionValues:p.values?.map(f=>({name:f}))||[]}))));let i=r instanceof Array?r:hydrogenReact.flattenConnection(r),{searchParams:u,path:l,alreadyOnProductPage:y}=da(e,o,n),g=c.filter(p=>p?.optionValues?.length===1),d=a?a?.selectedOptions?.reduce((p,f)=>(p[f.name]=f.value,p),{}):{};return react.createElement(react.Fragment,null,...react.useMemo(()=>c.map(p=>{let f,P=[];for(let m of p.optionValues){let C=new URLSearchParams(y?u:void 0);C.set(p.name,m.name),g.forEach(S=>{S.optionValues[0].name&&C.set(S.name,S.optionValues[0].name);});let A=i.find(S=>S?.selectedOptions?.every(M=>(C.get(M?.name)||d?.[M?.name])===M?.value)),T=u.get(p.name);!T&&a&&(T=d?.[p.name]||null);let h=T?T===m.name:!1;h&&(f=m.name);let I="?"+C.toString();P.push({value:m.name,optionValue:m,isAvailable:A?A.availableForSale:!0,to:l+I,search:I,isActive:h,variant:A});}return s({option:{name:p.name,value:f,values:P}})}),[c,i,s]))}var pa=e=>{if(typeof e?.url>"u")throw new TypeError(`Expected a Request instance, got ${typeof e}`);let t=new URL(e.url).searchParams,r=[];return t.forEach((o,n)=>{r.push({name:n,value:o});}),r};function da(e,t,r){let{pathname:o,search:n}=react$1.useLocation(),a=react$1.useNavigation();return react.useMemo(()=>{let s=/(\/[a-zA-Z]{2}-[a-zA-Z]{2}\/)/g.exec(o),c=s&&s.length>0;t=t.startsWith("/")?t.substring(1):t;let i=c?`${s[0]}${t}/${e}`:`/${t}/${e}`;return {searchParams:new URLSearchParams(r||a.state!=="loading"?n:a.location.search),alreadyOnProductPage:i===o,path:i}},[o,n,r,e,t,a])}function ma(e,t){let r=react$1.useNavigation(),[o,n]=react.useState([]);if(react.useEffect(()=>{Promise.resolve(t).then(a=>{a&&n(a instanceof Array?a:a.product?.variants?.nodes||[]);}).catch(a=>{reportError(new Error("[h2:error:useOptimisticVariant] An error occurred while resolving the variants for the optimistic product hook.",{cause:a}));});},[JSON.stringify(t)]),r.state==="loading"){let a=new URLSearchParams(r.location.search),s=!1,c=o.find(i=>i.selectedOptions?i.selectedOptions.every(u=>a.get(u.name)===u.value):(s||(s=!0,reportError(new Error("[h2:error:useOptimisticVariant] The optimistic product hook requires your product query to include variants with the selectedOptions field."))),!1));if(c)return {...c,isOptimistic:!0}}return e}var to=react.createContext(void 0),Pa=to.Provider,tr=()=>react.useContext(to);function Aa(e){let t=Ze(),r=Ta(t,e);return {nonce:t,header:r,NonceProvider:({children:n})=>react.createElement(Pa,{value:t},n)}}function Ta(e,t){let{shop:r,...o}=t??{},n=`'nonce-${e}'`,a=["'self'","'unsafe-inline'","https://cdn.shopify.com"],s=["'self'","https://monorail-edge.shopifysvc.com"];r&&r.checkoutDomain&&s.push(`https://${r.checkoutDomain}`),r&&r.storeDomain&&s.push(`https://${r.storeDomain}`);let i={baseUri:["'self'"],defaultSrc:["'self'",n,"https://cdn.shopify.com","https://shopify.com"],frameAncestors:["'none'"],styleSrc:a,connectSrc:s},u=Object.assign({},i,o);for(let l in i){let y=o[l];l&&y&&(u[l]=va(y,i[l]));}return u.scriptSrc instanceof Array?u.scriptSrc=[...u.scriptSrc.filter(l=>!l.startsWith("'nonce")),n]:u.defaultSrc instanceof Array&&(u.defaultSrc=[...u.defaultSrc.filter(l=>!l.startsWith("'nonce")),n]),Sa__default.default({directives:u})}function va(e,t){let r=typeof t=="string"?[t]:t,o=Array.isArray(e)?e:[String(e)];return Array.isArray(r)?r.every(a=>a==="'none'")?o:[...o,...r]:r}var Ra=react.forwardRef((e,t)=>{let{waitForHydration:r,src:o,...n}=e;if(r)return jsxRuntime.jsx(ba,{src:o,options:n});let a=tr();return jsxRuntime.jsx("script",{suppressHydrationWarning:!0,...n,src:o,nonce:a,ref:t})});function ba({src:e,options:t}){if(!e)throw new Error("`waitForHydration` with the Script component requires a `src` prop");return hydrogenReact.useLoadScript(e,{attributes:t}),null}function Oa(e){let t=react$1.useFetchers(),r={};for(let{formData:o}of t)if(o?.get("optimistic-identifier")===e)try{if(o.has("optimistic-data")){let n=JSON.parse(String(o.get("optimistic-data")));Object.assign(r,n);}}catch{}return r}function xa({id:e,data:t}){return jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[jsxRuntime.jsx("input",{type:"hidden",name:"optimistic-identifier",value:e}),jsxRuntime.jsx("input",{type:"hidden",name:"optimistic-data",value:JSON.stringify(t)})]})}function ka(e){return jsxRuntime.jsx(hydrogenReact.ShopPayButton,{channel:"hydrogen",...e})}function Ce(e){let{type:t,data:r={},customData:o}=e,n=react$1.useLocation(),{publish:a,cart:s,prevCart:c,shop:i,customData:u}=X(),l=n.pathname+n.search,y={...r,customData:{...u,...o},cart:s,prevCart:c,shop:i};return react.useEffect(()=>{i?.shopId&&(y={...y,url:window.location.href},a(t,y));},[a,l,i?.shopId]),null}function no(e){return jsxRuntime.jsx(Ce,{...e,type:"page_viewed"})}function ao(e){return jsxRuntime.jsx(Ce,{...e,type:"product_viewed"})}function so(e){return jsxRuntime.jsx(Ce,{...e,type:"collection_viewed"})}function io(e){return jsxRuntime.jsx(Ce,{...e,type:"cart_viewed"})}function co(e){return jsxRuntime.jsx(Ce,{...e,type:"search_viewed"})}function uo(e){return jsxRuntime.jsx(Ce,{...e})}var H={PAGE_VIEWED:"page_viewed",PRODUCT_VIEWED:"product_viewed",COLLECTION_VIEWED:"collection_viewed",CART_VIEWED:"cart_viewed",SEARCH_VIEWED:"search_viewed",CART_UPDATED:"cart_updated",PRODUCT_ADD_TO_CART:"product_added_to_cart",PRODUCT_REMOVED_FROM_CART:"product_removed_from_cart",CUSTOM_EVENT:"custom_"};var Ba="https://cdn.shopify.com/shopifycloud/consent-tracking-api/v0.2/consent-tracking-api.js",Ga="https://cdn.shopify.com/shopifycloud/privacy-banner/storefront-banner.js";function or(e){console.error(`[h2:error:useCustomerPrivacy] Unable to setup Customer Privacy API: Missing consent.${e} configuration.`);}function nr(e){let{withPrivacyBanner:t=!1,onVisitorConsentCollected:r,onReady:o,checkoutDomain:n,storefrontAccessToken:a,country:s,locale:c,sameDomainForStorefrontApi:i}=e,u=react.useMemo(()=>i??Ar(),[i]),l=react.useMemo(()=>u&&!Tr(),[u]),y=hydrogenReact.useShopifyCookies({fetchTrackingValues:l,storefrontAccessToken:a,ignoreDeprecatedCookies:!0}),g=react.useMemo(hydrogenReact.getTrackingValues,[y]),{revalidate:d}=react$1.useRevalidator();hydrogenReact.useLoadScript(t?Ga:Ba,{attributes:{id:"customer-privacy-api"}});let{observing:p,setLoaded:f,apisLoaded:P}=Wa({withPrivacyBanner:t}),m=react.useMemo(()=>{n||or("checkoutDomain"),a||or("storefrontAccessToken"),(a.startsWith("shpat_")||a.length!==32)&&console.error("[h2:error:useCustomerPrivacy] It looks like you passed a private access token, make sure to use the public token");let A=ja(n);return {checkoutRootDomain:u&&typeof window<"u"?window.location.host:n,storefrontRootDomain:A?"."+A:void 0,storefrontAccessToken:a,country:s,locale:c}},[or,n,a,s,c]);react.useEffect(()=>{let A=T=>{let h=hydrogenReact.getTrackingValues();if((g.visitToken!==h.visitToken||g.uniqueToken!==h.uniqueToken)&&d(),r){let I=se();if(I?.shouldShowBanner()){let S=I.currentVisitorConsent();if(S){let M="";if(S.marketing===M&&S.analytics===M&&S.preferences===M)return}}r(T.detail);}};return document.addEventListener("visitorConsentCollected",A),()=>{document.removeEventListener("visitorConsentCollected",A);}},[r]),react.useEffect(()=>{if(!t||p.current.privacyBanner)return;p.current.privacyBanner=!0;let A=window.privacyBanner||void 0;Object.defineProperty(window,"privacyBanner",{configurable:!0,get(){return A},set(h){typeof h=="object"&&h!==null&&"showPreferences"in h&&"loadBanner"in h&&(A=yo({privacyBanner:h,config:m}),f.privacyBanner());}});},[t,m,yo,f.privacyBanner]),react.useEffect(()=>{if(p.current.customerPrivacy)return;p.current.customerPrivacy=!0;let A=null,T=window.Shopify||void 0;Object.defineProperty(window,"Shopify",{configurable:!0,get(){return T},set(h){typeof h=="object"&&h!==null&&Object.keys(h).length===0&&(T=h,Object.defineProperty(window.Shopify,"customerPrivacy",{configurable:!0,get(){return A},set(I){if(typeof I=="object"&&I!==null&&"setTrackingConsent"in I){let S=I;A={...S,setTrackingConsent:lo({customerPrivacy:S,config:m})},T={...T,customerPrivacy:A},f.customerPrivacy();}}}));}});},[m,lo,f.customerPrivacy]),react.useEffect(()=>{if(!P||!y)return;let A=se();if(A&&!A.cachedConsent){let T=hydrogenReact.getTrackingValues();T.consent&&(A.cachedConsent=T.consent);}if(t){let T=Oe();T&&T.loadBanner(m);}Qa(),o?.();},[P,y]);let C={customerPrivacy:se()};return t&&(C.privacyBanner=Oe()),C}var po=!1;function Qa(){if(po)return;po=!0;let e=new CustomEvent("shopifyCustomerPrivacyApiLoaded");document.dispatchEvent(e);}function Wa({withPrivacyBanner:e}){let t=react.useRef({customerPrivacy:!1,privacyBanner:!1}),[r,o]=react.useState(e?[!1,!1]:[!1]),n=r.every(Boolean);return {observing:t,setLoaded:{customerPrivacy:()=>{o(e?s=>[!0,s[1]]:()=>[!0]);},privacyBanner:()=>{e&&o(s=>[s[0],!0]);}},apisLoaded:n}}function ja(e){if(typeof window>"u")return;let t=window.location.host,r=e.split(".").reverse(),o=t.split(".").reverse(),n=[];return r.forEach((a,s)=>{a===o[s]&&n.push(a);}),n.reverse().join(".")||void 0}function lo({customerPrivacy:e,config:t}){let r=e.setTrackingConsent,{locale:o,country:n,...a}=t;function s(c,i){r({...a,headlessStorefront:!0,...c},i);}return s}function yo({privacyBanner:e,config:t}){let r=e.loadBanner,o=e.showPreferences;function n(s){if(typeof s=="object"){r({...t,...s});return}r(t);}function a(s){if(typeof s=="object"){o({...t,...s});return}o(t);}return {loadBanner:n,showPreferences:a}}function se(){try{return window.Shopify&&window.Shopify.customerPrivacy?window.Shopify?.customerPrivacy:null}catch{return null}}function Oe(){try{return window&&window?.privacyBanner?window.privacyBanner:null}catch{return null}}var fo="2024.10.2";function Za(){let e=se();if(!e)throw new Error("Shopify Customer Privacy API not available. Must be used within a useEffect. Make sure to load the Shopify Customer Privacy API with useCustomerPrivacy() or <AnalyticsProvider>.");return e}function go({consent:e,onReady:t,domain:r}){let{subscribe:o,register:n,canTrack:a}=X(),[s,c]=react.useState(!1),[i,u]=react.useState(!1),[l,y]=react.useState(""),g=react.useRef(!1),{checkoutDomain:d,storefrontAccessToken:p,language:f}=e,{ready:P}=n("Internal_Shopify_Analytics");nr({...e,locale:f,checkoutDomain:d||"mock.shop",storefrontAccessToken:p||"abcdefghijklmnopqrstuvwxyz123456",onReady:()=>!e.withPrivacyBanner&&u(!0),onVisitorConsentCollected:C=>{try{y(JSON.stringify(C));}catch{}u(!0);}});let m=react.useMemo(()=>i?a():!0,[i,a,l]);return hydrogenReact.useShopifyCookies({hasUserConsent:m,domain:r,checkoutDomain:d,fetchTrackingValues:!1,ignoreDeprecatedCookies:!i}),react.useEffect(()=>{g.current||(g.current=!0,o(H.PAGE_VIEWED,ts),o(H.PRODUCT_VIEWED,rs),o(H.COLLECTION_VIEWED,os),o(H.SEARCH_VIEWED,ns),o(H.PRODUCT_ADD_TO_CART,as),c(!0));},[o]),react.useEffect(()=>{s&&i&&(P(),t());},[s,i,t]),null}function nt(e){console.error(`[h2:error:ShopifyAnalytics] Unable to send Shopify analytics: Missing shop.${e} configuration.`);}function _e(e){let t=Za(),r=t.analyticsProcessingAllowed();if(!e?.shop?.shopId){nt("shopId");return}if(!e?.shop?.acceptedLanguage){nt("acceptedLanguage");return}if(!e?.shop?.currency){nt("currency");return}if(!e?.shop?.hydrogenSubchannelId){nt("hydrogenSubchannelId");return}return {shopifySalesChannel:"hydrogen",assetVersionId:fo,...e.shop,hasUserConsent:r,...hydrogenReact.getClientBrowserParameters(),analyticsAllowed:t.analyticsProcessingAllowed(),marketingAllowed:t.marketingAllowed(),saleOfDataAllowed:t.saleOfDataAllowed(),ccpaEnforced:!t.saleOfDataAllowed(),gdprEnforced:!(t.marketingAllowed()&&t.analyticsProcessingAllowed())}}function es(e,t){if(t===null)return;let r=_e(e);return r?{...r,cartId:t.id}:void 0}var ie={};function ts(e){let t=_e(e);t&&(hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.PAGE_VIEW_2,payload:{...t,...ie}}),ie={});}function rs(e){let t=_e(e);if(t&&ho({type:"product",products:e.products})){let r=sr(e.products);ie={pageType:hydrogenReact.AnalyticsPageType.product,resourceId:r[0].productGid},t={...t,...ie,products:sr(e.products)},hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.PRODUCT_VIEW,payload:t});}}function os(e){let t=_e(e);t&&(ie={pageType:hydrogenReact.AnalyticsPageType.collection,resourceId:e.collection.id},t={...t,...ie,collectionHandle:e.collection.handle,collectionId:e.collection.id},hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.COLLECTION_VIEW,payload:t}));}function ns(e){let t=_e(e);t&&(ie={pageType:hydrogenReact.AnalyticsPageType.search},t={...t,...ie,searchString:e.searchTerm},hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.SEARCH_VIEW,payload:t}));}function as(e){let{cart:t,currentLine:r}=e,o=es(e,t);!o||!r?.id||ss({matchedLine:r,eventPayload:o});}function ss({matchedLine:e,eventPayload:t}){let r={id:e.merchandise.product.id,variantId:e.merchandise.id,title:e.merchandise.product.title,variantTitle:e.merchandise.title,vendor:e.merchandise.product.vendor,price:e.merchandise.price.amount,quantity:e.quantity,productType:e.merchandise.product.productType,sku:e.merchandise.sku};ho({type:"cart",products:[r]})&&hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.ADD_TO_CART,payload:{...t,products:sr([r])}});}function de(e,t,r,o){if(e==="cart"){let n=`${r?"merchandise":"merchandise.product"}.${t}`;console.error(`[h2:error:ShopifyAnalytics] Can't set up cart analytics events because the \`cart.lines[].${n}\` value is missing from your GraphQL cart query. In your project, search for where \`fragment CartLine on CartLine\` is defined and make sure \`${n}\` is part of your cart query. Check the Hydrogen Skeleton template for reference: https://github.com/Shopify/hydrogen/blob/main/templates/skeleton/app/lib/fragments.ts#L25-L56.`);}else {let n=`${o||t}`;console.error(`[h2:error:ShopifyAnalytics] Can't set up product view analytics events because the \`${n}\` is missing from your \`<Analytics.ProductView>\`. Make sure \`${n}\` is part of your products data prop. Check the Hydrogen Skeleton template for reference: https://github.com/Shopify/hydrogen/blob/main/templates/skeleton/app/routes/products.%24handle.tsx#L159-L165.`);}}function ho({type:e,products:t}){return !t||t.length===0?(de(e,"",!1,"data.products"),!1):(t.forEach(r=>{if(!r.id)return de(e,"id",!1),!1;if(!r.title)return de(e,"title",!1),!1;if(!r.price)return de(e,"price.amount",!0,"price"),!1;if(!r.vendor)return de(e,"vendor",!1),!1;if(!r.variantId)return de(e,"id",!0,"variantId"),!1;if(!r.variantTitle)return de(e,"title",!0,"variantTitle"),!1}),!0)}function sr(e){return e.map(t=>{let r={productGid:t.id,variantGid:t.variantId,name:t.title,variantName:t.variantTitle,brand:t.vendor,price:t.price,quantity:t.quantity||1,category:t.productType};return t.sku&&(r.sku=t.sku),t.productType&&(r.category=t.productType),r})}function Po(e){console.error(`[h2:error:CartAnalytics] Can't set up cart analytics events because the \`cart.${e}\` value is missing from your GraphQL cart query. In your project, search for where \`fragment CartApiQuery on Cart\` is defined and make sure \`${e}\` is part of your cart query. Check the Hydrogen Skeleton template for reference: https://github.com/Shopify/hydrogen/blob/main/templates/skeleton/app/lib/fragments.ts#L59.`);}function Ao({cart:e,setCarts:t}){let{publish:r,shop:o,customData:n,canTrack:a,cart:s,prevCart:c}=X(),i=react.useRef(null);return react.useEffect(()=>{if(e)return Promise.resolve(e).then(u=>{if(u&&u.lines){if(!u.id){Po("id");return}if(!u.updatedAt){Po("updatedAt");return}}t(({cart:l,prevCart:y})=>u?.updatedAt!==l?.updatedAt?{cart:u,prevCart:l}:{cart:l,prevCart:y});}),()=>{}},[t,e]),react.useEffect(()=>{if(!s||!s?.updatedAt||s?.updatedAt===c?.updatedAt)return;let u;try{u=JSON.parse(localStorage.getItem("cartLastUpdatedAt")||"");}catch{u=null;}if(s.id===u?.id&&s.updatedAt===u?.updatedAt)return;let l={eventTimestamp:Date.now(),cart:s,prevCart:c,shop:o,customData:n};if(s.updatedAt===i.current)return;i.current=s.updatedAt,r("cart_updated",l),localStorage.setItem("cartLastUpdatedAt",JSON.stringify({id:s.id,updatedAt:s.updatedAt}));let y=c?.lines?hydrogenReact.flattenConnection(c?.lines):[],g=s.lines?hydrogenReact.flattenConnection(s.lines):[];y?.forEach(d=>{let p=g.filter(f=>d.id===f.id);if(p?.length===1){let f=p[0];d.quantity<f.quantity?r("product_added_to_cart",{...l,prevLine:d,currentLine:f}):d.quantity>f.quantity&&r("product_removed_from_cart",{...l,prevLine:d,currentLine:f});}else r("product_removed_from_cart",{...l,prevLine:d});}),g?.forEach(d=>{let p=y.filter(f=>d.id===f.id);(!p||p.length===0)&&r("product_added_to_cart",{...l,currentLine:d});});},[s,c,r,o,n,a]),null}var ls="https://cdn.shopify.com/shopifycloud/perf-kit/shopify-perf-kit-1.0.1.min.js";function To({shop:e}){let t=react.useRef(!1),{subscribe:r,register:o}=X(),{ready:n}=o("Internal_Shopify_Perf_Kit"),a=hydrogenReact.useLoadScript(ls,{attributes:{id:"perfkit","data-application":"hydrogen","data-shop-id":hydrogenReact.parseGid(e.shopId).id.toString(),"data-storefront-id":e.hydrogenSubchannelId,"data-monorail-region":"global","data-spa-mode":"true","data-resource-timing-sampling-rate":"100"}});return react.useEffect(()=>{a!=="done"||t.current||(t.current=!0,r(H.PAGE_VIEWED,()=>{window.PerfKit?.navigate();}),r(H.PRODUCT_VIEWED,()=>{window.PerfKit?.setPageType("product");}),r(H.COLLECTION_VIEWED,()=>{window.PerfKit?.setPageType("collection");}),r(H.SEARCH_VIEWED,()=>{window.PerfKit?.setPageType("search");}),r(H.CART_VIEWED,()=>{window.PerfKit?.setPageType("cart");}),n());},[r,n,a]),null}var hs={canTrack:()=>!1,cart:null,customData:{},prevCart:null,publish:()=>{},shop:null,subscribe:()=>{},register:()=>({ready:()=>{}}),customerPrivacy:null,privacyBanner:null},wo=react.createContext(hs),st=new Map,ke={};function Oo(){return Object.values(ke).every(Boolean)}function vo(e,t){st.has(e)||st.set(e,new Map),st.get(e)?.set(t.toString(),t);}var it=new Map;function Io(e,t){if(!Oo()){it.set(e,t);return}xo(e,t);}function xo(e,t){(st.get(e)??new Map).forEach((r,o)=>{try{r(t);}catch(n){typeof n=="object"&&n instanceof Error?console.error("Analytics publish error",n.message,o,n.stack):console.error("Analytics publish error",n,o);}});}function Eo(e){return ke.hasOwnProperty(e)||(ke[e]=!1),{ready:()=>{ke[e]=!0,Oo()&&it.size>0&&(it.forEach((t,r)=>{xo(r,t);}),it.clear());}}}function Ro(){try{return window.Shopify.customerPrivacy.analyticsProcessingAllowed()}catch{}return !1}function bo(e,t){return `[h2:error:Analytics.Provider] - ${e} is required. Make sure ${t} is defined in your environment variables. See https://h2o.fyi/analytics/consent to learn how to setup environment variables in the Shopify admin.`}function Cs({canTrack:e,cart:t,children:r,consent:o,customData:n={},shop:a=null,cookieDomain:s}){let{shop:c}=Ss(a),[i,u]=react.useState(!!e),[l,y]=react.useState(!1),[g,d]=react.useState({cart:null,prevCart:null}),[p,f]=react.useState(e?()=>e:()=>Ro);if(c)if(/\/68817551382$/.test(c.shopId))re("[h2:error:Analytics.Provider] - Mock shop is used. Analytics will not work properly.");else {if(!o.checkoutDomain){let m=bo("consent.checkoutDomain","PUBLIC_CHECKOUT_DOMAIN");Tt(m);}if(!o.storefrontAccessToken){let m=bo("consent.storefrontAccessToken","PUBLIC_STOREFRONT_API_TOKEN");Tt(m);}o?.country||(o.country="US"),o?.language||(o.language="EN"),o.withPrivacyBanner===void 0&&(o.withPrivacyBanner=!1);}let P=react.useMemo(()=>({canTrack:p,...g,customData:n,publish:p()?Io:()=>{},shop:c,subscribe:vo,register:Eo,customerPrivacy:se(),privacyBanner:Oe()}),[i,p,g,g.cart?.updatedAt,g.prevCart,Io,vo,n,c,Eo,JSON.stringify(ke),se,Oe]);return jsxRuntime.jsxs(wo.Provider,{value:P,children:[r,!!c&&jsxRuntime.jsx(no,{}),!!c&&!!t&&jsxRuntime.jsx(Ao,{cart:t,setCarts:d}),!!c&&jsxRuntime.jsx(go,{consent:o,onReady:()=>{u(!0),f(e?()=>e:()=>Ro),y(!0);},domain:s}),!!c&&l&&jsxRuntime.jsx(To,{shop:c})]})}function X(){let e=react.useContext(wo);if(!e)throw new Error("[h2:error:useAnalytics] 'useAnalytics()' must be a descendent of <AnalyticsProvider/>");return e}function Ss(e){let[t,r]=react.useState(null);return react.useEffect(()=>(Promise.resolve(e).then(r),()=>{}),[r,e]),{shop:t}}async function Ps({storefront:e,publicStorefrontId:t="0"}){return e.query(As,{cache:e.CacheLong()}).then(({shop:r,localization:o})=>({shopId:r.id,acceptedLanguage:o.language.isoCode,currency:o.country.currency.isoCode,hydrogenSubchannelId:t}))}var As=`#graphql
  query ShopData(
    $country: CountryCode
    $language: LanguageCode
  ) @inContext(country: $country, language: $language) {
    shop {
      id
    }
    localization {
      country {
        currency {
          isoCode
        }
      }
      language {
        isoCode
      }
    }
  }
`,Ts={CartView:io,CollectionView:so,CustomView:uo,ProductView:ao,Provider:Cs,SearchView:co};var Rs=function(e){return jsxRuntime.jsx(hydrogenReact.RichText,{...e,components:{link:({node:t})=>jsxRuntime.jsx(react$1.Link,{to:t.url,title:t.title,target:t.target,prefetch:"intent",children:t.children}),...e.components}})};function bs(e){let{env:t,request:r,cache:o,waitUntil:n,i18n:a,session:s,logErrors:c,storefront:i={},customerAccount:u,cart:l={}}=e;s||console.warn("[h2:warn:createHydrogenContext] A session object is required to create hydrogen context.");let{storefront:y}=br({cache:o,waitUntil:n,i18n:a,logErrors:c,storefrontHeaders:i.headers||ws(r),storefrontApiVersion:i.apiVersion,storefrontId:t.PUBLIC_STOREFRONT_ID,storeDomain:t.PUBLIC_STORE_DOMAIN,privateStorefrontToken:t.PRIVATE_STOREFRONT_API_TOKEN,publicStorefrontToken:t.PUBLIC_STOREFRONT_API_TOKEN}),g=Mt({session:s,request:r,waitUntil:n,logErrors:c,customerApiVersion:u?.apiVersion,authUrl:u?.authUrl,customAuthStatusHandler:u?.customAuthStatusHandler,unstableB2b:u?.unstableB2b,customerAccountId:t.PUBLIC_CUSTOMER_ACCOUNT_API_CLIENT_ID,shopId:t.SHOP_ID,customerAccountUrl:t.PUBLIC_CUSTOMER_ACCOUNT_API_URL}),d=er({getCartId:l.getId||Xt(r.headers),setCartId:l.setId||Zt(),cartQueryFragment:l.queryFragment,cartMutateFragment:l.mutateFragment,customMethods:l.customMethods,storefront:y,customerAccount:g});return {storefront:y,customerAccount:g,cart:d,env:t,waitUntil:n,session:s}}function ws(e){return {requestGroupId:B(e,"request-id"),buyerIp:B(e,"oxygen-buyer-ip"),buyerIpSig:B(e,"X-Shopify-Client-IP-Sig"),cookie:B(e,"cookie"),purpose:B(e,"sec-purpose")||B(e,"purpose")}}function xs({build:e,mode:t,poweredByHeader:r=!0,getLoadContext:o,collectTrackingInformation:n=!0,proxyStandardRoutes:a=!0}){let s=serverRuntime.createRequestHandler(e,t),c=r?i=>i.headers.append("powered-by","Shopify, Hydrogen"):void 0;return async i=>{let u=i.method;if((u==="GET"||u==="HEAD")&&i.body)return new Response(`${u} requests cannot have a body`,{status:400});let l=new URL(i.url);if(l.pathname.includes("//"))return new Response(null,{status:301,headers:{location:l.pathname.replace(/\/+/g,"/")}});let y=o?await o(i):void 0,g=y?.storefront;if(a&&(g||re("[h2:createRequestHandler] Storefront instance is required to proxy standard routes."),g?.isStorefrontApiUrl(i))){let p=await g.forward(i);return c?.(p),p}let d=await s(i,y);if(g&&a){n&&g.setCollectedSubrequestHeaders(d);let p=i.headers.get("sec-fetch-dest");(p&&p==="document"||i.headers.get("accept")?.includes("text/html"))&&Ie(d,{[Ge]:"1"});}return c?.(d),d}}var Ds=`<?xml version="1.0" encoding="UTF-8"?>
<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
`,_s=`
</sitemapindex>`,_o=`<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">`,Uo="</urlset>";async function Us(e){let{storefront:t,request:r,types:o=["products","pages","collections","metaObjects","articles","blogs"],customChildSitemaps:n=[]}=e;if(!r||!r.url)throw new Error("A request object is required to generate a sitemap index");if(!t||!t.query)throw new Error("A storefront client is required to generate a sitemap index");let a=await t.query(Gs);if(!a)throw console.warn("[h2:sitemap:warning] Sitemap index is available in API version 2024-10 and later"),new Response("Sitemap index not found.",{status:404});let s=new URL(r.url).origin,c=Ds+o.map(i=>{if(!a[i])throw new Error(`[h2:sitemap:error] No data found for type ${i}. Check types passed to \`getSitemapIndex\``);return Ls(i,a[i].pagesCount.count,s)}).join(`
`)+n.map(i=>"  <sitemap><loc>"+(s+(i.startsWith("/")?i:"/"+i))+"</loc></sitemap>").join(`
`)+_s;return new Response(c,{headers:{"Content-Type":"application/xml","Cache-Control":`max-age=${60*60*24}`}})}async function ks(e){let{storefront:t,request:r,params:o,getLink:n,locales:a=[],getChangeFreq:s,noItemsFallback:c="/"}=e;if(!o)throw new Error("[h2:sitemap:error] Remix params object is required to generate a sitemap");if(!r||!r.url)throw new Error("A request object is required to generate a sitemap");if(!t||!t.query)throw new Error("A storefront client is required to generate a index");if(!n)throw new Error("A `getLink` function to generate each resource is required to build a sitemap");if(!o.type||!o.page)throw new Response("No data found",{status:404});let i=o.type,u=Qs[i];if(!u)throw new Response("Not found",{status:404});let l=await t.query(u,{variables:{page:parseInt(o.page,10)}});if(!l)throw console.warn("[h2:sitemap:warning] Sitemap is available in API version 2024-10 and later"),new Response("Sitemap not found.",{status:404});let y=new URL(r.url).origin,g="";return l?.sitemap?.resources?.items?.length?g=_o+l.sitemap.resources.items.map(d=>Ns({getChangeFreq:s,url:n({type:d.type??i,baseUrl:y,handle:d.handle}),type:i,getLink:n,updatedAt:d.updatedAt,handle:d.handle,metaobjectType:d.type,locales:a,baseUrl:y})).join(`
`)+Uo:g=_o+`
  <url><loc>${y+c}</loc></url>
`+Uo,new Response(g,{headers:{"Content-Type":"application/xml","Cache-Control":`max-age=${60*60*24}`}})}function Ls(e,t,r){let o="";for(let n=1;n<=t;n++)o+=`  <sitemap><loc>${r}/sitemap/${e}/${n}.xml</loc></sitemap>
`;return o}function Ns({url:e,updatedAt:t,locales:r,type:o,getLink:n,baseUrl:a,handle:s,getChangeFreq:c,metaobjectType:i}){return `<url>
  <loc>${e}</loc>
  <lastmod>${t}</lastmod>
  <changefreq>${c?c({type:i??o,handle:s}):"weekly"}</changefreq>
${r.map(u=>Vs(n({type:i??o,baseUrl:a,handle:s,locale:u}),u)).join(`
`)}
</url>
  `.trim()}function Vs(e,t){return `  <xhtml:link rel="alternate" hreflang="${t}" href="${e}" />`}var Ms=`#graphql
    query SitemapProducts($page: Int!) {
      sitemap(type: PRODUCT) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,Hs=`#graphql
    query SitemapCollections($page: Int!) {
      sitemap(type: COLLECTION) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,Fs=`#graphql
    query SitemapArticles($page: Int!) {
      sitemap(type: ARTICLE) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,qs=`#graphql
    query SitemapPages($page: Int!) {
      sitemap(type: PAGE) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,$s=`#graphql
    query SitemapBlogs($page: Int!) {
      sitemap(type: BLOG) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,Bs=`#graphql
    query SitemapMetaobjects($page: Int!) {
      sitemap(type: METAOBJECT) {
        resources(page: $page) {
          items {
            handle
            updatedAt
            ... on SitemapResourceMetaobject {
              type
            }
          }
        }
      }
    }
`,Gs=`#graphql
query SitemapIndex {
  products: sitemap(type: PRODUCT) {
    pagesCount {
      count
    }
  }
  collections: sitemap(type: COLLECTION) {
    pagesCount {
      count
    }
  }
  articles: sitemap(type: ARTICLE) {
    pagesCount {
      count
    }
  }
  pages: sitemap(type: PAGE) {
    pagesCount {
      count
    }
  }
  blogs: sitemap(type: BLOG) {
    pagesCount {
      count
    }
  }
  metaObjects: sitemap(type: METAOBJECT) {
    pagesCount {
      count
    }
  }
}
`,Qs={products:Ms,articles:Fs,collections:Hs,pages:qs,blogs:$s,metaObjects:Bs};//! @see: https://shopify.dev/docs/api/storefront/latest/mutations/cartCreate
//! @see https://shopify.dev/docs/api/storefront/latest/queries/cart
//! @see: https://shopify.dev/docs/api/storefront/latest/mutations/cartLinesAdd
//! @see: https://shopify.dev/docs/api/storefront/latest/mutations/cartLinesUpdate
//! @see: https://shopify.dev/docs/api/storefront/latest/mutations/cartLinesRemove
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartDiscountCodesUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartGiftCardCodesUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartBuyerIdentityUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartNoteUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartSelectedDeliveryOptionsUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartMetafieldsSet
//! @see https://shopify.dev/docs/api/storefront/2024-10/mutations/cartMetafieldDelete

Object.defineProperty(exports, "AnalyticsEventName", {
  enumerable: true,
  get: function () { return hydrogenReact.AnalyticsEventName; }
});
Object.defineProperty(exports, "AnalyticsPageType", {
  enumerable: true,
  get: function () { return hydrogenReact.AnalyticsPageType; }
});
Object.defineProperty(exports, "ExternalVideo", {
  enumerable: true,
  get: function () { return hydrogenReact.ExternalVideo; }
});
Object.defineProperty(exports, "IMAGE_FRAGMENT", {
  enumerable: true,
  get: function () { return hydrogenReact.IMAGE_FRAGMENT; }
});
Object.defineProperty(exports, "Image", {
  enumerable: true,
  get: function () { return hydrogenReact.Image; }
});
Object.defineProperty(exports, "MediaFile", {
  enumerable: true,
  get: function () { return hydrogenReact.MediaFile; }
});
Object.defineProperty(exports, "ModelViewer", {
  enumerable: true,
  get: function () { return hydrogenReact.ModelViewer; }
});
Object.defineProperty(exports, "Money", {
  enumerable: true,
  get: function () { return hydrogenReact.Money; }
});
Object.defineProperty(exports, "ShopifySalesChannel", {
  enumerable: true,
  get: function () { return hydrogenReact.ShopifySalesChannel; }
});
Object.defineProperty(exports, "Video", {
  enumerable: true,
  get: function () { return hydrogenReact.Video; }
});
Object.defineProperty(exports, "customerAccountApiCustomScalars", {
  enumerable: true,
  get: function () { return hydrogenReact.customerAccountApiCustomScalars; }
});
Object.defineProperty(exports, "decodeEncodedVariant", {
  enumerable: true,
  get: function () { return hydrogenReact.decodeEncodedVariant; }
});
Object.defineProperty(exports, "flattenConnection", {
  enumerable: true,
  get: function () { return hydrogenReact.flattenConnection; }
});
Object.defineProperty(exports, "getAdjacentAndFirstAvailableVariants", {
  enumerable: true,
  get: function () { return hydrogenReact.getAdjacentAndFirstAvailableVariants; }
});
Object.defineProperty(exports, "getClientBrowserParameters", {
  enumerable: true,
  get: function () { return hydrogenReact.getClientBrowserParameters; }
});
Object.defineProperty(exports, "getProductOptions", {
  enumerable: true,
  get: function () { return hydrogenReact.getProductOptions; }
});
Object.defineProperty(exports, "getShopifyCookies", {
  enumerable: true,
  get: function () { return hydrogenReact.getShopifyCookies; }
});
Object.defineProperty(exports, "getTrackingValues", {
  enumerable: true,
  get: function () { return hydrogenReact.getTrackingValues; }
});
Object.defineProperty(exports, "isOptionValueCombinationInEncodedVariant", {
  enumerable: true,
  get: function () { return hydrogenReact.isOptionValueCombinationInEncodedVariant; }
});
Object.defineProperty(exports, "mapSelectedProductOptionToObject", {
  enumerable: true,
  get: function () { return hydrogenReact.mapSelectedProductOptionToObject; }
});
Object.defineProperty(exports, "parseGid", {
  enumerable: true,
  get: function () { return hydrogenReact.parseGid; }
});
Object.defineProperty(exports, "parseMetafield", {
  enumerable: true,
  get: function () { return hydrogenReact.parseMetafield; }
});
Object.defineProperty(exports, "sendShopifyAnalytics", {
  enumerable: true,
  get: function () { return hydrogenReact.sendShopifyAnalytics; }
});
Object.defineProperty(exports, "storefrontApiCustomScalars", {
  enumerable: true,
  get: function () { return hydrogenReact.storefrontApiCustomScalars; }
});
Object.defineProperty(exports, "useLoadScript", {
  enumerable: true,
  get: function () { return hydrogenReact.useLoadScript; }
});
Object.defineProperty(exports, "useMoney", {
  enumerable: true,
  get: function () { return hydrogenReact.useMoney; }
});
Object.defineProperty(exports, "useSelectedOptionInUrlParam", {
  enumerable: true,
  get: function () { return hydrogenReact.useSelectedOptionInUrlParam; }
});
Object.defineProperty(exports, "useShopifyCookies", {
  enumerable: true,
  get: function () { return hydrogenReact.useShopifyCookies; }
});
exports.Analytics = Ts;
exports.AnalyticsEvent = H;
exports.CacheCustom = ft;
exports.CacheLong = yt;
exports.CacheNone = dt;
exports.CacheShort = ce;
exports.CartForm = J;
exports.InMemoryCache = bt;
exports.OptimisticInput = xa;
exports.Pagination = vn;
exports.RichText = Rs;
exports.Script = Ra;
exports.Seo = Sn;
exports.ShopPayButton = ka;
exports.VariantSelector = ua;
exports.cartAttributesUpdateDefault = Yt;
exports.cartBuyerIdentityUpdateDefault = Wt;
exports.cartCreateDefault = Ht;
exports.cartDiscountCodesUpdateDefault = Gt;
exports.cartGetDefault = Ft;
exports.cartGetIdDefault = Xt;
exports.cartGiftCardCodesUpdateDefault = Qt;
exports.cartLinesAddDefault = qt;
exports.cartLinesRemoveDefault = Bt;
exports.cartLinesUpdateDefault = $t;
exports.cartMetafieldDeleteDefault = Jt;
exports.cartMetafieldsSetDefault = zt;
exports.cartNoteUpdateDefault = jt;
exports.cartSelectedDeliveryOptionsUpdateDefault = Kt;
exports.cartSetIdDefault = Zt;
exports.changelogHandler = Ln;
exports.createCartHandler = er;
exports.createContentSecurityPolicy = Aa;
exports.createCustomerAccountClient = Mt;
exports.createHydrogenContext = bs;
exports.createRequestHandler = xs;
exports.createStorefrontClient = br;
exports.createWithCache = rn;
exports.formatAPIResult = R;
exports.generateCacheControlHeader = Pe;
exports.getPaginationVariables = En;
exports.getSelectedProductOptions = pa;
exports.getSeoMeta = Pn;
exports.getShopAnalytics = Ps;
exports.getSitemap = ks;
exports.getSitemapIndex = Us;
exports.graphiqlLoader = sn;
exports.storefrontRedirect = nn;
exports.useAnalytics = X;
exports.useCustomerPrivacy = nr;
exports.useNonce = tr;
exports.useOptimisticCart = oa;
exports.useOptimisticData = Oa;
exports.useOptimisticVariant = ma;
